// NewPromotion/Access/App.js
/**
 * Created with CodeModel.
 * User: 田林
 * Date: 2013-09-20
 * Time: 12:16:12
 * To change this template use File | Settings | File Templates.
 */
var client = require ( '../MySql' );
var Tools = require ( '../Tools' );
var EventEmitter = require ( 'events' ).EventEmitter;
var event = new EventEmitter ();

/*-----获取记录数--------------*/
var GetRsCount=function(Filter,CallBack){
    var Sql = "Select Count(*) as Count From App Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

var GetRsCount_Parms=function(Filter,Values ,CallBack){
    var Sql = "Select Count(*) as Count From App Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

/**
 * 功能:获取记录数
 *   重载方法说明：
 *   参数化查询
 *   function(String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *   非参数化查询
 *   function(String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.GetRsCount = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        GetRsCount ( arguments[0] ,arguments[1]);
        return;
    }
    if ( Tools.CheckArgs ( [String ,Object, Function] , arguments ) ) {
        GetRsCount_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/App.js=>GetRsCount 缺少参数");
    CallBack(false,null);
}
/*-----统计求和--------------*/
var Sum=function(Field,Filter,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From App Where "+Filter+"";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

var Sum_Parms=function(Field,Filter,Values ,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From App Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

/**
 * 功能：统计字段数值
 * 重载方法说明：
 * 参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *    非参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.Sum = function () {
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        Sum ( arguments[0] ,arguments[1],arguments[2]);
        return;
    }
    if ( Tools.CheckArgs ( [String, String ,Object, Function] , arguments ) ) {
        Sum_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/App.js=>Sum 缺少参数");
    CallBack(false,null);
}

/*-----查询所有记录--------------*/
var SelectAll = function ( CallBack ) {
    var Sql = "Select * From App";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectAll_Field = function (Field, CallBack ) {
    var Sql = "Select "+Field+" From App";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


var SelectAllByOrder = function (Field ,  Order , CallBack ) {
    var Sql = "Select "+Field+" From App  Order By " + Order;
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 功能: 查询所有记录
 *   重载方法说明：
 *   获取所有记录
 *   function(回调函数Function（err,Result）)
 *   获取所有记录 - 指定要返回的字段
 *   function(String 要返回的字段（Field1，Field2） ，回调函数Function（err,Result）)
 *   获取所有记录 - 排序
 *   function(String 要返回的字段（Field1，Field2） , String 排序表达式（Filed Desc），回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectAll = function () {
    if ( Tools.CheckArgs ( [Function] , arguments ) ) {
        SelectAll ( arguments[0] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        SelectAll_Field ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        SelectAllByOrder ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/App.js=>SelectAll 缺少参数");
    CallBack(false,null);
}
/*-----根据关键字查询记录--------------*/

var Select = function ( AppIndex , CallBack ) {
    var Sql = "Select * From App Where AppIndex=?AppIndex";
    var Values = {};
    Values.AppIndex = AppIndex;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

    var Select_Field = function (Field, AppIndex , CallBack ) {
    var Sql = "Select "+Field+" From App Where AppIndex=?AppIndex";
    var Values = {};
    Values.AppIndex = AppIndex;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

/**
 * 功能: 根据主键查询记录
 *   重载方法说明：
 *   返回所有字段  - 主键值为字符串
 *   function(String 主键值 ，回调函数Function（err,Result）)
 *   返回所有字段   - 主键值为数字
 *   function(Number 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段  - 主键值为字符串
 *   function(String 要返回的字段（Field1，Field2） ，String 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段 - 主键值为数字
 *   function(String 要返回的字段（Field1，Field2） ，Number 主键值 ，回调函数Function（err,Result）)
 * @constructor
 */
exports.Select = function () {

    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [Number, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Number, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/App.js=>Select 缺少参数");
    CallBack(false,null);
}

/*-----根据条件查询记录--------------*/
var SelectByFilter = function ( Filter , CallBack ) {
    var Sql = "Select * From App Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field = function (Field, Filter , CallBack ) {
    var Sql = "Select "+Field+" From App Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Parms = function ( Filter , Values , CallBack ) {
    var Sql = "Select * From App Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field_Parms = function (Field, Filter , Values , CallBack ) {
    var Sql = "Select "+Field+" From App Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 功能: 根据条件查询记录
 *   重载方法说明：
 *   返回所有字段 - 非参数化查询
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   返回指定字段 - 非参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ，回调函数Function（err,Result）)
 *   返回所有字段  - 参数化查询
 *   function(String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ， 回调函数Function（err,Result）)
 *   返回指定的字段 - 参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectByFilter = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        SelectByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        SelectByFilter_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Object , Function] , arguments ) ) {
        SelectByFilter_Parms ( arguments[0] , arguments[1] , arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Object , Function] , arguments ) ) {
        SelectByFilter_Field_Parms ( arguments[0] , arguments[1] , arguments[2],arguments[3] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/App.js=>SelectByFilter 缺少参数");
    CallBack(false,null);
}

/*-----插入--------------*/
/**
 * 插入记录
 * @param Model 实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Insert = function ( Model , CallBack ) {
    var Sql = "Insert Into App Set AppUser=?AppUser,AppName=?AppName,SpreaderID=?SpreaderID,AppKey=?AppKey,AppCode=?AppCode,AppLabel=?AppLabel,AppType=?AppType,AppCount=?AppCount,DayAppCount=?DayAppCount,AppErro=?AppErro,AppIsSpider=?AppIsSpider,AppAble=?AppAble,AppWtime=?AppWtime,SiteSign=?SiteSign,AppErroMsg=?AppErroMsg,AccessToken=?AccessToken ";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 插入记录
 * @param ModelList 实体集合 ，Type:Object, 实例：[{key1:value1,key2:value2},{key1:value1,key2:value2}]
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.InsertList = function ( ModelList , CallBack ) {
    var Sql = "Insert Into App Set AppUser=?AppUser,AppName=?AppName,SpreaderID=?SpreaderID,AppKey=?AppKey,AppCode=?AppCode,AppLabel=?AppLabel,AppType=?AppType,AppCount=?AppCount,DayAppCount=?DayAppCount,AppErro=?AppErro,AppIsSpider=?AppIsSpider,AppAble=?AppAble,AppWtime=?AppWtime,SiteSign=?SiteSign,AppErroMsg=?AppErroMsg,AccessToken=?AccessToken";
    client.GetConnection ( function ( err , connection ) {
        if ( err ) {
            CallBack ( err , null );
            return;
        }
        for ( var i = 0 ; i < ModelList.length ; i ++ ) {
            var Model = ModelList[0];
            client.Query ( connection , Sql , Model , function ( err ) {
                if ( err ) {
                    event.emit ( 'InsertFail' );
                    return;
                }
                if ( i == ModelList.length - 1 ) {
                    event.emit ( 'InsertSuccess' );
                }
            } )
        }
        event.on ( 'InsertSuccess' , function () {
            connection.release ();
            CallBack ( true , null );
        } )
        event.on ( 'InsertFail' , function () {
            connection.release ();
            CallBack ( false , null );
        } )
    } )


}

/*-----更新--------------*/
/**
 * 更新记录
 * @param Model   实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Update = function ( Model , CallBack ) {
    var Sql = "Update App Set  AppUser=?AppUser,AppName=?AppName,SpreaderID=?SpreaderID,AppKey=?AppKey,AppCode=?AppCode,AppLabel=?AppLabel,AppType=?AppType,AppCount=?AppCount,DayAppCount=?DayAppCount,AppErro=?AppErro,AppIsSpider=?AppIsSpider,AppAble=?AppAble,AppWtime=?AppWtime,SiteSign=?SiteSign,AppErroMsg=?AppErroMsg,AccessToken=?AccessToken Where AppIndex=?AppIndex";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 更新字段值 AppUser
 * @param Key  主键值 Key:AppIndex
 * @param Value AppUser的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AppUser = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AppUser=?AppUser where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AppName
 * @param Key  主键值 Key:AppIndex
 * @param Value AppName的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AppName = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AppName=?AppName where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SpreaderID
 * @param Key  主键值 Key:AppIndex
 * @param Value SpreaderID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SpreaderID = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set SpreaderID=?SpreaderID where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AppKey
 * @param Key  主键值 Key:AppIndex
 * @param Value AppKey的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AppKey = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AppKey=?AppKey where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AppCode
 * @param Key  主键值 Key:AppIndex
 * @param Value AppCode的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AppCode = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AppCode=?AppCode where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AppLabel
 * @param Key  主键值 Key:AppIndex
 * @param Value AppLabel的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AppLabel = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AppLabel=?AppLabel where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AppType
 * @param Key  主键值 Key:AppIndex
 * @param Value AppType的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AppType = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AppType=?AppType where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AppCount
 * @param Key  主键值 Key:AppIndex
 * @param Value AppCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AppCount = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AppCount=?AppCount where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 DayAppCount
 * @param Key  主键值 Key:AppIndex
 * @param Value DayAppCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_DayAppCount = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set DayAppCount=?DayAppCount where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AppErro
 * @param Key  主键值 Key:AppIndex
 * @param Value AppErro的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AppErro = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AppErro=?AppErro where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AppIsSpider
 * @param Key  主键值 Key:AppIndex
 * @param Value AppIsSpider的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AppIsSpider = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AppIsSpider=?AppIsSpider where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AppAble
 * @param Key  主键值 Key:AppIndex
 * @param Value AppAble的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AppAble = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AppAble=?AppAble where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AppWtime
 * @param Key  主键值 Key:AppIndex
 * @param Value AppWtime的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AppWtime = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AppWtime=?AppWtime where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SiteSign
 * @param Key  主键值 Key:AppIndex
 * @param Value SiteSign的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SiteSign = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set SiteSign=?SiteSign where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AppErroMsg
 * @param Key  主键值 Key:AppIndex
 * @param Value AppErroMsg的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AppErroMsg = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AppErroMsg=?AppErroMsg where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AccessToken
 * @param Key  主键值 Key:AppIndex
 * @param Value AccessToken的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AccessToken = function ( AppIndex , Value , CallBack ) {
    var Sql = "Update App Set AccessToken=?AccessToken where AppIndex=?AppIndex";
    var Model = {};
    Model.Value = Value;
    Model.AppIndex = AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/*-----删除数据--------------*/
/**
 *  根据主键删除数据
 * @param Key     主键值
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.Delete=function(AppIndex,CallBack){
    var Sql = "Delete From App Where AppIndex=?AppIndex";
    var Model={};
    Model.AppIndex=AppIndex;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *   删除所有数据
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.DeleteAll=function(CallBack){
    var Sql = "Delete From App";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter=function(Filter,CallBack){
    var Sql = "Delete From App where "+Filter+"";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter_Parms=function(Filter,Values,CallBack){
    var Sql = "Delete From App where "+Filter+"";
    client.Query ( Sql  ,Values, function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *    删除所有记录
 *    重载方法说明：
 *   根据条件删除 - 非参数化
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   根据条件删除 - 参数化
 *   function(String 查询条件，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports. DeleteByFilter=function(){
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        DeleteByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Object, Function] , arguments ) ) {
        DeleteByFilter_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/App.js=>DeleteByFilter 缺少参数");
    CallBack(false,null);
}


/*-----获取分页数据--------------*/
var GetPageRow = function ( CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from App  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field = function ( Field , CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from App  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Parms = function ( CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from App  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field_Parms = function ( Field , CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from App  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


/**
 *    获取分页数据
 *    重载方法说明：
 *   获取所有字段
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *   获取指定字段
 *   function(String 要查询的字段（Filed1,Filed2）, Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *    获取所有字段  - 参数化查询
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ，Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ， 回调函数Function（err,Result）)
 *   获取指定字段 - 参数化查询
 *   function(String 要查询的字段（Filed1,Filed2）,  Number 当前页 ，Number 分页尺寸 , String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.GetPageRow=function(){
    if ( Tools.CheckArgs ( [Number , Number , String , Function] , arguments ) ) {
        GetPageRow ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String , Function] , arguments ) ) {
        GetPageRow_Field ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [ Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Field_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4],arguments[5] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/App.js=>GetPageRow 缺少参数");
    CallBack(false,null);
}
