// NewPromotion/Access/ClubThread.js
/**
 * Created with CodeModel.
 * User: 田林
 * Date: 2013-09-20
 * Time: 12:16:12
 * To change this template use File | Settings | File Templates.
 */
var client = require ( '../MySql' );
var Tools = require ( '../Tools' );
var EventEmitter = require ( 'events' ).EventEmitter;
var event = new EventEmitter ();

/*-----获取记录数--------------*/
var GetRsCount=function(Filter,CallBack){
    var Sql = "Select Count(*) as Count From ClubThread Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

var GetRsCount_Parms=function(Filter,Values ,CallBack){
    var Sql = "Select Count(*) as Count From ClubThread Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

/**
 * 功能:获取记录数
 *   重载方法说明：
 *   参数化查询
 *   function(String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *   非参数化查询
 *   function(String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.GetRsCount = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        GetRsCount ( arguments[0] ,arguments[1]);
        return;
    }
    if ( Tools.CheckArgs ( [String ,Object, Function] , arguments ) ) {
        GetRsCount_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubThread.js=>GetRsCount 缺少参数");
    CallBack(false,null);
}
/*-----统计求和--------------*/
var Sum=function(Field,Filter,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From ClubThread Where "+Filter+"";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

var Sum_Parms=function(Field,Filter,Values ,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From ClubThread Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

/**
 * 功能：统计字段数值
 * 重载方法说明：
 * 参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *    非参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.Sum = function () {
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        Sum ( arguments[0] ,arguments[1],arguments[2]);
        return;
    }
    if ( Tools.CheckArgs ( [String, String ,Object, Function] , arguments ) ) {
        Sum_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubThread.js=>Sum 缺少参数");
    CallBack(false,null);
}

/*-----查询所有记录--------------*/
var SelectAll = function ( CallBack ) {
    var Sql = "Select * From ClubThread";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectAll_Field = function (Field, CallBack ) {
    var Sql = "Select "+Field+" From ClubThread";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


var SelectAllByOrder = function (Field ,  Order , CallBack ) {
    var Sql = "Select "+Field+" From ClubThread  Order By " + Order;
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 功能: 查询所有记录
 *   重载方法说明：
 *   获取所有记录
 *   function(回调函数Function（err,Result）)
 *   获取所有记录 - 指定要返回的字段
 *   function(String 要返回的字段（Field1，Field2） ，回调函数Function（err,Result）)
 *   获取所有记录 - 排序
 *   function(String 要返回的字段（Field1，Field2） , String 排序表达式（Filed Desc），回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectAll = function () {
    if ( Tools.CheckArgs ( [Function] , arguments ) ) {
        SelectAll ( arguments[0] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        SelectAll_Field ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        SelectAllByOrder ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubThread.js=>SelectAll 缺少参数");
    CallBack(false,null);
}
/*-----根据关键字查询记录--------------*/

var Select = function ( ThreadID , CallBack ) {
    var Sql = "Select * From ClubThread Where ThreadID=?ThreadID";
    var Values = {};
    Values.ThreadID = ThreadID;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

    var Select_Field = function (Field, ThreadID , CallBack ) {
    var Sql = "Select "+Field+" From ClubThread Where ThreadID=?ThreadID";
    var Values = {};
    Values.ThreadID = ThreadID;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

/**
 * 功能: 根据主键查询记录
 *   重载方法说明：
 *   返回所有字段  - 主键值为字符串
 *   function(String 主键值 ，回调函数Function（err,Result）)
 *   返回所有字段   - 主键值为数字
 *   function(Number 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段  - 主键值为字符串
 *   function(String 要返回的字段（Field1，Field2） ，String 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段 - 主键值为数字
 *   function(String 要返回的字段（Field1，Field2） ，Number 主键值 ，回调函数Function（err,Result）)
 * @constructor
 */
exports.Select = function () {

    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [Number, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Number, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubThread.js=>Select 缺少参数");
    CallBack(false,null);
}

/*-----根据条件查询记录--------------*/
var SelectByFilter = function ( Filter , CallBack ) {
    var Sql = "Select * From ClubThread Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field = function (Field, Filter , CallBack ) {
    var Sql = "Select "+Field+" From ClubThread Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Parms = function ( Filter , Values , CallBack ) {
    var Sql = "Select * From ClubThread Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field_Parms = function (Field, Filter , Values , CallBack ) {
    var Sql = "Select "+Field+" From ClubThread Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 功能: 根据条件查询记录
 *   重载方法说明：
 *   返回所有字段 - 非参数化查询
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   返回指定字段 - 非参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ，回调函数Function（err,Result）)
 *   返回所有字段  - 参数化查询
 *   function(String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ， 回调函数Function（err,Result）)
 *   返回指定的字段 - 参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectByFilter = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        SelectByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        SelectByFilter_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Object , Function] , arguments ) ) {
        SelectByFilter_Parms ( arguments[0] , arguments[1] , arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Object , Function] , arguments ) ) {
        SelectByFilter_Field_Parms ( arguments[0] , arguments[1] , arguments[2],arguments[3] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubThread.js=>SelectByFilter 缺少参数");
    CallBack(false,null);
}

/*-----插入--------------*/
/**
 * 插入记录
 * @param Model 实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Insert = function ( Model , CallBack ) {
    var Sql = "Insert Into ClubThread Set Thread_SN=?Thread_SN,Title=?Title,UserID=?UserID,UserName=?UserName,ModuleID=?ModuleID,LabelID=?LabelID,ThreadType=?ThreadType,GroupID=?GroupID,GroupLabelID=?GroupLabelID,ReadCount=?ReadCount,ReqCount=?ReqCount,AgreeCount=?AgreeCount,CollectionCount=?CollectionCount,ForWardCount=?ForWardCount,SupportCount=?SupportCount,NoSupportCount=?NoSupportCount,LastReqUserID=?LastReqUserID,LastReqUserName=?LastReqUserName,ThreadAble=?ThreadAble,IsRecommend=?IsRecommend,IsTop=?IsTop,CreateTime=?CreateTime,UpdateTime=?UpdateTime,ThreadContent=?ThreadContent,ThreadImg=?ThreadImg,ThreadProdID=?ThreadProdID,ThreadProdImg=?ThreadProdImg,ThreadProdType=?ThreadProdType ";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 插入记录
 * @param ModelList 实体集合 ，Type:Object, 实例：[{key1:value1,key2:value2},{key1:value1,key2:value2}]
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.InsertList = function ( ModelList , CallBack ) {
    var Sql = "Insert Into ClubThread Set Thread_SN=?Thread_SN,Title=?Title,UserID=?UserID,UserName=?UserName,ModuleID=?ModuleID,LabelID=?LabelID,ThreadType=?ThreadType,GroupID=?GroupID,GroupLabelID=?GroupLabelID,ReadCount=?ReadCount,ReqCount=?ReqCount,AgreeCount=?AgreeCount,CollectionCount=?CollectionCount,ForWardCount=?ForWardCount,SupportCount=?SupportCount,NoSupportCount=?NoSupportCount,LastReqUserID=?LastReqUserID,LastReqUserName=?LastReqUserName,ThreadAble=?ThreadAble,IsRecommend=?IsRecommend,IsTop=?IsTop,CreateTime=?CreateTime,UpdateTime=?UpdateTime,ThreadContent=?ThreadContent,ThreadImg=?ThreadImg,ThreadProdID=?ThreadProdID,ThreadProdImg=?ThreadProdImg,ThreadProdType=?ThreadProdType";
    client.GetConnection ( function ( err , connection ) {
        if ( err ) {
            CallBack ( err , null );
            return;
        }
        for ( var i = 0 ; i < ModelList.length ; i ++ ) {
            var Model = ModelList[0];
            client.Query ( connection , Sql , Model , function ( err ) {
                if ( err ) {
                    event.emit ( 'InsertFail' );
                    return;
                }
                if ( i == ModelList.length - 1 ) {
                    event.emit ( 'InsertSuccess' );
                }
            } )
        }
        event.on ( 'InsertSuccess' , function () {
            connection.release ();
            CallBack ( true , null );
        } )
        event.on ( 'InsertFail' , function () {
            connection.release ();
            CallBack ( false , null );
        } )
    } )


}

/*-----更新--------------*/
/**
 * 更新记录
 * @param Model   实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Update = function ( Model , CallBack ) {
    var Sql = "Update ClubThread Set  Thread_SN=?Thread_SN,Title=?Title,UserID=?UserID,UserName=?UserName,ModuleID=?ModuleID,LabelID=?LabelID,ThreadType=?ThreadType,GroupID=?GroupID,GroupLabelID=?GroupLabelID,ReadCount=?ReadCount,ReqCount=?ReqCount,AgreeCount=?AgreeCount,CollectionCount=?CollectionCount,ForWardCount=?ForWardCount,SupportCount=?SupportCount,NoSupportCount=?NoSupportCount,LastReqUserID=?LastReqUserID,LastReqUserName=?LastReqUserName,ThreadAble=?ThreadAble,IsRecommend=?IsRecommend,IsTop=?IsTop,CreateTime=?CreateTime,UpdateTime=?UpdateTime,ThreadContent=?ThreadContent,ThreadImg=?ThreadImg,ThreadProdID=?ThreadProdID,ThreadProdImg=?ThreadProdImg,ThreadProdType=?ThreadProdType Where ThreadID=?ThreadID";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 更新字段值 Thread_SN
 * @param Key  主键值 Key:ThreadID
 * @param Value Thread_SN的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_Thread_SN = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set Thread_SN=?Thread_SN where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 Title
 * @param Key  主键值 Key:ThreadID
 * @param Value Title的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_Title = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set Title=?Title where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserID
 * @param Key  主键值 Key:ThreadID
 * @param Value UserID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserID = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set UserID=?UserID where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserName
 * @param Key  主键值 Key:ThreadID
 * @param Value UserName的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserName = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set UserName=?UserName where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ModuleID
 * @param Key  主键值 Key:ThreadID
 * @param Value ModuleID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ModuleID = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set ModuleID=?ModuleID where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 LabelID
 * @param Key  主键值 Key:ThreadID
 * @param Value LabelID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_LabelID = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set LabelID=?LabelID where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ThreadType
 * @param Key  主键值 Key:ThreadID
 * @param Value ThreadType的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ThreadType = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set ThreadType=?ThreadType where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 GroupID
 * @param Key  主键值 Key:ThreadID
 * @param Value GroupID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_GroupID = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set GroupID=?GroupID where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 GroupLabelID
 * @param Key  主键值 Key:ThreadID
 * @param Value GroupLabelID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_GroupLabelID = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set GroupLabelID=?GroupLabelID where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ReadCount
 * @param Key  主键值 Key:ThreadID
 * @param Value ReadCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ReadCount = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set ReadCount=?ReadCount where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ReqCount
 * @param Key  主键值 Key:ThreadID
 * @param Value ReqCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ReqCount = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set ReqCount=?ReqCount where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AgreeCount
 * @param Key  主键值 Key:ThreadID
 * @param Value AgreeCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AgreeCount = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set AgreeCount=?AgreeCount where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 CollectionCount
 * @param Key  主键值 Key:ThreadID
 * @param Value CollectionCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_CollectionCount = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set CollectionCount=?CollectionCount where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ForWardCount
 * @param Key  主键值 Key:ThreadID
 * @param Value ForWardCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ForWardCount = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set ForWardCount=?ForWardCount where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SupportCount
 * @param Key  主键值 Key:ThreadID
 * @param Value SupportCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SupportCount = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set SupportCount=?SupportCount where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 NoSupportCount
 * @param Key  主键值 Key:ThreadID
 * @param Value NoSupportCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_NoSupportCount = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set NoSupportCount=?NoSupportCount where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 LastReqUserID
 * @param Key  主键值 Key:ThreadID
 * @param Value LastReqUserID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_LastReqUserID = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set LastReqUserID=?LastReqUserID where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 LastReqUserName
 * @param Key  主键值 Key:ThreadID
 * @param Value LastReqUserName的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_LastReqUserName = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set LastReqUserName=?LastReqUserName where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ThreadAble
 * @param Key  主键值 Key:ThreadID
 * @param Value ThreadAble的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ThreadAble = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set ThreadAble=?ThreadAble where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 IsRecommend
 * @param Key  主键值 Key:ThreadID
 * @param Value IsRecommend的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_IsRecommend = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set IsRecommend=?IsRecommend where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 IsTop
 * @param Key  主键值 Key:ThreadID
 * @param Value IsTop的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_IsTop = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set IsTop=?IsTop where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 CreateTime
 * @param Key  主键值 Key:ThreadID
 * @param Value CreateTime的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_CreateTime = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set CreateTime=?CreateTime where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UpdateTime
 * @param Key  主键值 Key:ThreadID
 * @param Value UpdateTime的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UpdateTime = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set UpdateTime=?UpdateTime where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ThreadContent
 * @param Key  主键值 Key:ThreadID
 * @param Value ThreadContent的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ThreadContent = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set ThreadContent=?ThreadContent where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ThreadImg
 * @param Key  主键值 Key:ThreadID
 * @param Value ThreadImg的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ThreadImg = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set ThreadImg=?ThreadImg where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ThreadProdID
 * @param Key  主键值 Key:ThreadID
 * @param Value ThreadProdID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ThreadProdID = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set ThreadProdID=?ThreadProdID where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ThreadProdImg
 * @param Key  主键值 Key:ThreadID
 * @param Value ThreadProdImg的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ThreadProdImg = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set ThreadProdImg=?ThreadProdImg where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ThreadProdType
 * @param Key  主键值 Key:ThreadID
 * @param Value ThreadProdType的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ThreadProdType = function ( ThreadID , Value , CallBack ) {
    var Sql = "Update ClubThread Set ThreadProdType=?ThreadProdType where ThreadID=?ThreadID";
    var Model = {};
    Model.Value = Value;
    Model.ThreadID = ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/*-----删除数据--------------*/
/**
 *  根据主键删除数据
 * @param Key     主键值
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.Delete=function(ThreadID,CallBack){
    var Sql = "Delete From ClubThread Where ThreadID=?ThreadID";
    var Model={};
    Model.ThreadID=ThreadID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *   删除所有数据
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.DeleteAll=function(CallBack){
    var Sql = "Delete From ClubThread";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter=function(Filter,CallBack){
    var Sql = "Delete From ClubThread where "+Filter+"";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter_Parms=function(Filter,Values,CallBack){
    var Sql = "Delete From ClubThread where "+Filter+"";
    client.Query ( Sql  ,Values, function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *    删除所有记录
 *    重载方法说明：
 *   根据条件删除 - 非参数化
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   根据条件删除 - 参数化
 *   function(String 查询条件，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports. DeleteByFilter=function(){
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        DeleteByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Object, Function] , arguments ) ) {
        DeleteByFilter_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubThread.js=>DeleteByFilter 缺少参数");
    CallBack(false,null);
}


/*-----获取分页数据--------------*/
var GetPageRow = function ( CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from ClubThread  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field = function ( Field , CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from ClubThread  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Parms = function ( CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from ClubThread  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field_Parms = function ( Field , CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from ClubThread  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


/**
 *    获取分页数据
 *    重载方法说明：
 *   获取所有字段
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *   获取指定字段
 *   function(String 要查询的字段（Filed1,Filed2）, Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *    获取所有字段  - 参数化查询
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ，Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ， 回调函数Function（err,Result）)
 *   获取指定字段 - 参数化查询
 *   function(String 要查询的字段（Filed1,Filed2）,  Number 当前页 ，Number 分页尺寸 , String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.GetPageRow=function(){
    if ( Tools.CheckArgs ( [Number , Number , String , Function] , arguments ) ) {
        GetPageRow ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String , Function] , arguments ) ) {
        GetPageRow_Field ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [ Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Field_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4],arguments[5] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubThread.js=>GetPageRow 缺少参数");
    CallBack(false,null);
}
