// NewPromotion/Access/ClubModule.js
/**
 * Created with CodeModel.
 * User: 田林
 * Date: 2013-09-20
 * Time: 12:16:12
 * To change this template use File | Settings | File Templates.
 */
var client = require ( '../MySql' );
var Tools = require ( '../Tools' );
var EventEmitter = require ( 'events' ).EventEmitter;
var event = new EventEmitter ();

/*-----获取记录数--------------*/
var GetRsCount=function(Filter,CallBack){
    var Sql = "Select Count(*) as Count From ClubModule Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

var GetRsCount_Parms=function(Filter,Values ,CallBack){
    var Sql = "Select Count(*) as Count From ClubModule Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

/**
 * 功能:获取记录数
 *   重载方法说明：
 *   参数化查询
 *   function(String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *   非参数化查询
 *   function(String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.GetRsCount = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        GetRsCount ( arguments[0] ,arguments[1]);
        return;
    }
    if ( Tools.CheckArgs ( [String ,Object, Function] , arguments ) ) {
        GetRsCount_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubModule.js=>GetRsCount 缺少参数");
    CallBack(false,null);
}
/*-----统计求和--------------*/
var Sum=function(Field,Filter,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From ClubModule Where "+Filter+"";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

var Sum_Parms=function(Field,Filter,Values ,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From ClubModule Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

/**
 * 功能：统计字段数值
 * 重载方法说明：
 * 参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *    非参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.Sum = function () {
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        Sum ( arguments[0] ,arguments[1],arguments[2]);
        return;
    }
    if ( Tools.CheckArgs ( [String, String ,Object, Function] , arguments ) ) {
        Sum_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubModule.js=>Sum 缺少参数");
    CallBack(false,null);
}

/*-----查询所有记录--------------*/
var SelectAll = function ( CallBack ) {
    var Sql = "Select * From ClubModule";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectAll_Field = function (Field, CallBack ) {
    var Sql = "Select "+Field+" From ClubModule";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


var SelectAllByOrder = function (Field ,  Order , CallBack ) {
    var Sql = "Select "+Field+" From ClubModule  Order By " + Order;
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 功能: 查询所有记录
 *   重载方法说明：
 *   获取所有记录
 *   function(回调函数Function（err,Result）)
 *   获取所有记录 - 指定要返回的字段
 *   function(String 要返回的字段（Field1，Field2） ，回调函数Function（err,Result）)
 *   获取所有记录 - 排序
 *   function(String 要返回的字段（Field1，Field2） , String 排序表达式（Filed Desc），回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectAll = function () {
    if ( Tools.CheckArgs ( [Function] , arguments ) ) {
        SelectAll ( arguments[0] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        SelectAll_Field ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        SelectAllByOrder ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubModule.js=>SelectAll 缺少参数");
    CallBack(false,null);
}
/*-----根据关键字查询记录--------------*/

var Select = function ( ClubModuleID , CallBack ) {
    var Sql = "Select * From ClubModule Where ClubModuleID=?ClubModuleID";
    var Values = {};
    Values.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

    var Select_Field = function (Field, ClubModuleID , CallBack ) {
    var Sql = "Select "+Field+" From ClubModule Where ClubModuleID=?ClubModuleID";
    var Values = {};
    Values.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

/**
 * 功能: 根据主键查询记录
 *   重载方法说明：
 *   返回所有字段  - 主键值为字符串
 *   function(String 主键值 ，回调函数Function（err,Result）)
 *   返回所有字段   - 主键值为数字
 *   function(Number 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段  - 主键值为字符串
 *   function(String 要返回的字段（Field1，Field2） ，String 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段 - 主键值为数字
 *   function(String 要返回的字段（Field1，Field2） ，Number 主键值 ，回调函数Function（err,Result）)
 * @constructor
 */
exports.Select = function () {

    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [Number, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Number, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubModule.js=>Select 缺少参数");
    CallBack(false,null);
}

/*-----根据条件查询记录--------------*/
var SelectByFilter = function ( Filter , CallBack ) {
    var Sql = "Select * From ClubModule Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field = function (Field, Filter , CallBack ) {
    var Sql = "Select "+Field+" From ClubModule Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Parms = function ( Filter , Values , CallBack ) {
    var Sql = "Select * From ClubModule Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field_Parms = function (Field, Filter , Values , CallBack ) {
    var Sql = "Select "+Field+" From ClubModule Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 功能: 根据条件查询记录
 *   重载方法说明：
 *   返回所有字段 - 非参数化查询
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   返回指定字段 - 非参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ，回调函数Function（err,Result）)
 *   返回所有字段  - 参数化查询
 *   function(String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ， 回调函数Function（err,Result）)
 *   返回指定的字段 - 参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectByFilter = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        SelectByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        SelectByFilter_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Object , Function] , arguments ) ) {
        SelectByFilter_Parms ( arguments[0] , arguments[1] , arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Object , Function] , arguments ) ) {
        SelectByFilter_Field_Parms ( arguments[0] , arguments[1] , arguments[2],arguments[3] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubModule.js=>SelectByFilter 缺少参数");
    CallBack(false,null);
}

/*-----插入--------------*/
/**
 * 插入记录
 * @param Model 实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Insert = function ( Model , CallBack ) {
    var Sql = "Insert Into ClubModule Set ModuleParentID=?ModuleParentID,IsParent=?IsParent,ModuleName=?ModuleName,ModuleThreadCount_today=?ModuleThreadCount_today,ModuleThreadCount=?ModuleThreadCount,ModuleReqThreadCount=?ModuleReqThreadCount,ModuleLabelCount=?ModuleLabelCount,ModuleLogo=?ModuleLogo,ModuleSort=?ModuleSort,ModuleReMark=?ModuleReMark,ModuleAble=?ModuleAble,ModuleAdminUser=?ModuleAdminUser ";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 插入记录
 * @param ModelList 实体集合 ，Type:Object, 实例：[{key1:value1,key2:value2},{key1:value1,key2:value2}]
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.InsertList = function ( ModelList , CallBack ) {
    var Sql = "Insert Into ClubModule Set ModuleParentID=?ModuleParentID,IsParent=?IsParent,ModuleName=?ModuleName,ModuleThreadCount_today=?ModuleThreadCount_today,ModuleThreadCount=?ModuleThreadCount,ModuleReqThreadCount=?ModuleReqThreadCount,ModuleLabelCount=?ModuleLabelCount,ModuleLogo=?ModuleLogo,ModuleSort=?ModuleSort,ModuleReMark=?ModuleReMark,ModuleAble=?ModuleAble,ModuleAdminUser=?ModuleAdminUser";
    client.GetConnection ( function ( err , connection ) {
        if ( err ) {
            CallBack ( err , null );
            return;
        }
        for ( var i = 0 ; i < ModelList.length ; i ++ ) {
            var Model = ModelList[0];
            client.Query ( connection , Sql , Model , function ( err ) {
                if ( err ) {
                    event.emit ( 'InsertFail' );
                    return;
                }
                if ( i == ModelList.length - 1 ) {
                    event.emit ( 'InsertSuccess' );
                }
            } )
        }
        event.on ( 'InsertSuccess' , function () {
            connection.release ();
            CallBack ( true , null );
        } )
        event.on ( 'InsertFail' , function () {
            connection.release ();
            CallBack ( false , null );
        } )
    } )


}

/*-----更新--------------*/
/**
 * 更新记录
 * @param Model   实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Update = function ( Model , CallBack ) {
    var Sql = "Update ClubModule Set  ModuleParentID=?ModuleParentID,IsParent=?IsParent,ModuleName=?ModuleName,ModuleThreadCount_today=?ModuleThreadCount_today,ModuleThreadCount=?ModuleThreadCount,ModuleReqThreadCount=?ModuleReqThreadCount,ModuleLabelCount=?ModuleLabelCount,ModuleLogo=?ModuleLogo,ModuleSort=?ModuleSort,ModuleReMark=?ModuleReMark,ModuleAble=?ModuleAble,ModuleAdminUser=?ModuleAdminUser Where ClubModuleID=?ClubModuleID";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 更新字段值 ModuleParentID
 * @param Key  主键值 Key:ClubModuleID
 * @param Value ModuleParentID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ModuleParentID = function ( ClubModuleID , Value , CallBack ) {
    var Sql = "Update ClubModule Set ModuleParentID=?ModuleParentID where ClubModuleID=?ClubModuleID";
    var Model = {};
    Model.Value = Value;
    Model.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 IsParent
 * @param Key  主键值 Key:ClubModuleID
 * @param Value IsParent的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_IsParent = function ( ClubModuleID , Value , CallBack ) {
    var Sql = "Update ClubModule Set IsParent=?IsParent where ClubModuleID=?ClubModuleID";
    var Model = {};
    Model.Value = Value;
    Model.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ModuleName
 * @param Key  主键值 Key:ClubModuleID
 * @param Value ModuleName的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ModuleName = function ( ClubModuleID , Value , CallBack ) {
    var Sql = "Update ClubModule Set ModuleName=?ModuleName where ClubModuleID=?ClubModuleID";
    var Model = {};
    Model.Value = Value;
    Model.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ModuleThreadCount_today
 * @param Key  主键值 Key:ClubModuleID
 * @param Value ModuleThreadCount_today的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ModuleThreadCount_today = function ( ClubModuleID , Value , CallBack ) {
    var Sql = "Update ClubModule Set ModuleThreadCount_today=?ModuleThreadCount_today where ClubModuleID=?ClubModuleID";
    var Model = {};
    Model.Value = Value;
    Model.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ModuleThreadCount
 * @param Key  主键值 Key:ClubModuleID
 * @param Value ModuleThreadCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ModuleThreadCount = function ( ClubModuleID , Value , CallBack ) {
    var Sql = "Update ClubModule Set ModuleThreadCount=?ModuleThreadCount where ClubModuleID=?ClubModuleID";
    var Model = {};
    Model.Value = Value;
    Model.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ModuleReqThreadCount
 * @param Key  主键值 Key:ClubModuleID
 * @param Value ModuleReqThreadCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ModuleReqThreadCount = function ( ClubModuleID , Value , CallBack ) {
    var Sql = "Update ClubModule Set ModuleReqThreadCount=?ModuleReqThreadCount where ClubModuleID=?ClubModuleID";
    var Model = {};
    Model.Value = Value;
    Model.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ModuleLabelCount
 * @param Key  主键值 Key:ClubModuleID
 * @param Value ModuleLabelCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ModuleLabelCount = function ( ClubModuleID , Value , CallBack ) {
    var Sql = "Update ClubModule Set ModuleLabelCount=?ModuleLabelCount where ClubModuleID=?ClubModuleID";
    var Model = {};
    Model.Value = Value;
    Model.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ModuleLogo
 * @param Key  主键值 Key:ClubModuleID
 * @param Value ModuleLogo的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ModuleLogo = function ( ClubModuleID , Value , CallBack ) {
    var Sql = "Update ClubModule Set ModuleLogo=?ModuleLogo where ClubModuleID=?ClubModuleID";
    var Model = {};
    Model.Value = Value;
    Model.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ModuleSort
 * @param Key  主键值 Key:ClubModuleID
 * @param Value ModuleSort的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ModuleSort = function ( ClubModuleID , Value , CallBack ) {
    var Sql = "Update ClubModule Set ModuleSort=?ModuleSort where ClubModuleID=?ClubModuleID";
    var Model = {};
    Model.Value = Value;
    Model.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ModuleReMark
 * @param Key  主键值 Key:ClubModuleID
 * @param Value ModuleReMark的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ModuleReMark = function ( ClubModuleID , Value , CallBack ) {
    var Sql = "Update ClubModule Set ModuleReMark=?ModuleReMark where ClubModuleID=?ClubModuleID";
    var Model = {};
    Model.Value = Value;
    Model.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ModuleAble
 * @param Key  主键值 Key:ClubModuleID
 * @param Value ModuleAble的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ModuleAble = function ( ClubModuleID , Value , CallBack ) {
    var Sql = "Update ClubModule Set ModuleAble=?ModuleAble where ClubModuleID=?ClubModuleID";
    var Model = {};
    Model.Value = Value;
    Model.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ModuleAdminUser
 * @param Key  主键值 Key:ClubModuleID
 * @param Value ModuleAdminUser的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ModuleAdminUser = function ( ClubModuleID , Value , CallBack ) {
    var Sql = "Update ClubModule Set ModuleAdminUser=?ModuleAdminUser where ClubModuleID=?ClubModuleID";
    var Model = {};
    Model.Value = Value;
    Model.ClubModuleID = ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/*-----删除数据--------------*/
/**
 *  根据主键删除数据
 * @param Key     主键值
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.Delete=function(ClubModuleID,CallBack){
    var Sql = "Delete From ClubModule Where ClubModuleID=?ClubModuleID";
    var Model={};
    Model.ClubModuleID=ClubModuleID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *   删除所有数据
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.DeleteAll=function(CallBack){
    var Sql = "Delete From ClubModule";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter=function(Filter,CallBack){
    var Sql = "Delete From ClubModule where "+Filter+"";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter_Parms=function(Filter,Values,CallBack){
    var Sql = "Delete From ClubModule where "+Filter+"";
    client.Query ( Sql  ,Values, function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *    删除所有记录
 *    重载方法说明：
 *   根据条件删除 - 非参数化
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   根据条件删除 - 参数化
 *   function(String 查询条件，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports. DeleteByFilter=function(){
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        DeleteByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Object, Function] , arguments ) ) {
        DeleteByFilter_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubModule.js=>DeleteByFilter 缺少参数");
    CallBack(false,null);
}


/*-----获取分页数据--------------*/
var GetPageRow = function ( CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from ClubModule  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field = function ( Field , CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from ClubModule  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Parms = function ( CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from ClubModule  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field_Parms = function ( Field , CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from ClubModule  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


/**
 *    获取分页数据
 *    重载方法说明：
 *   获取所有字段
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *   获取指定字段
 *   function(String 要查询的字段（Filed1,Filed2）, Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *    获取所有字段  - 参数化查询
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ，Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ， 回调函数Function（err,Result）)
 *   获取指定字段 - 参数化查询
 *   function(String 要查询的字段（Filed1,Filed2）,  Number 当前页 ，Number 分页尺寸 , String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.GetPageRow=function(){
    if ( Tools.CheckArgs ( [Number , Number , String , Function] , arguments ) ) {
        GetPageRow ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String , Function] , arguments ) ) {
        GetPageRow_Field ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [ Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Field_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4],arguments[5] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/ClubModule.js=>GetPageRow 缺少参数");
    CallBack(false,null);
}
