// NewPromotion/Access/SFJ.js
/**
 * Created with CodeModel.
 * User: 田林
 * Date: 2013-09-20
 * Time: 12:16:13
 * To change this template use File | Settings | File Templates.
 */
var client = require ( '../MySql' );
var Tools = require ( '../Tools' );
var EventEmitter = require ( 'events' ).EventEmitter;
var event = new EventEmitter ();

/*-----获取记录数--------------*/
var GetRsCount=function(Filter,CallBack){
    var Sql = "Select Count(*) as Count From SFJ Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

var GetRsCount_Parms=function(Filter,Values ,CallBack){
    var Sql = "Select Count(*) as Count From SFJ Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

/**
 * 功能:获取记录数
 *   重载方法说明：
 *   参数化查询
 *   function(String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *   非参数化查询
 *   function(String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.GetRsCount = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        GetRsCount ( arguments[0] ,arguments[1]);
        return;
    }
    if ( Tools.CheckArgs ( [String ,Object, Function] , arguments ) ) {
        GetRsCount_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/SFJ.js=>GetRsCount 缺少参数");
    CallBack(false,null);
}
/*-----统计求和--------------*/
var Sum=function(Field,Filter,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From SFJ Where "+Filter+"";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

var Sum_Parms=function(Field,Filter,Values ,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From SFJ Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

/**
 * 功能：统计字段数值
 * 重载方法说明：
 * 参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *    非参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.Sum = function () {
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        Sum ( arguments[0] ,arguments[1],arguments[2]);
        return;
    }
    if ( Tools.CheckArgs ( [String, String ,Object, Function] , arguments ) ) {
        Sum_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/SFJ.js=>Sum 缺少参数");
    CallBack(false,null);
}

/*-----查询所有记录--------------*/
var SelectAll = function ( CallBack ) {
    var Sql = "Select * From SFJ";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectAll_Field = function (Field, CallBack ) {
    var Sql = "Select "+Field+" From SFJ";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


var SelectAllByOrder = function (Field ,  Order , CallBack ) {
    var Sql = "Select "+Field+" From SFJ  Order By " + Order;
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 功能: 查询所有记录
 *   重载方法说明：
 *   获取所有记录
 *   function(回调函数Function（err,Result）)
 *   获取所有记录 - 指定要返回的字段
 *   function(String 要返回的字段（Field1，Field2） ，回调函数Function（err,Result）)
 *   获取所有记录 - 排序
 *   function(String 要返回的字段（Field1，Field2） , String 排序表达式（Filed Desc），回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectAll = function () {
    if ( Tools.CheckArgs ( [Function] , arguments ) ) {
        SelectAll ( arguments[0] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        SelectAll_Field ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        SelectAllByOrder ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SFJ.js=>SelectAll 缺少参数");
    CallBack(false,null);
}
/*-----根据关键字查询记录--------------*/

var Select = function ( SFJ_ID , CallBack ) {
    var Sql = "Select * From SFJ Where SFJ_ID=?SFJ_ID";
    var Values = {};
    Values.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

    var Select_Field = function (Field, SFJ_ID , CallBack ) {
    var Sql = "Select "+Field+" From SFJ Where SFJ_ID=?SFJ_ID";
    var Values = {};
    Values.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

/**
 * 功能: 根据主键查询记录
 *   重载方法说明：
 *   返回所有字段  - 主键值为字符串
 *   function(String 主键值 ，回调函数Function（err,Result）)
 *   返回所有字段   - 主键值为数字
 *   function(Number 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段  - 主键值为字符串
 *   function(String 要返回的字段（Field1，Field2） ，String 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段 - 主键值为数字
 *   function(String 要返回的字段（Field1，Field2） ，Number 主键值 ，回调函数Function（err,Result）)
 * @constructor
 */
exports.Select = function () {

    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [Number, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Number, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SFJ.js=>Select 缺少参数");
    CallBack(false,null);
}

/*-----根据条件查询记录--------------*/
var SelectByFilter = function ( Filter , CallBack ) {
    var Sql = "Select * From SFJ Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field = function (Field, Filter , CallBack ) {
    var Sql = "Select "+Field+" From SFJ Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Parms = function ( Filter , Values , CallBack ) {
    var Sql = "Select * From SFJ Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field_Parms = function (Field, Filter , Values , CallBack ) {
    var Sql = "Select "+Field+" From SFJ Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 功能: 根据条件查询记录
 *   重载方法说明：
 *   返回所有字段 - 非参数化查询
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   返回指定字段 - 非参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ，回调函数Function（err,Result）)
 *   返回所有字段  - 参数化查询
 *   function(String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ， 回调函数Function（err,Result）)
 *   返回指定的字段 - 参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectByFilter = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        SelectByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        SelectByFilter_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Object , Function] , arguments ) ) {
        SelectByFilter_Parms ( arguments[0] , arguments[1] , arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Object , Function] , arguments ) ) {
        SelectByFilter_Field_Parms ( arguments[0] , arguments[1] , arguments[2],arguments[3] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SFJ.js=>SelectByFilter 缺少参数");
    CallBack(false,null);
}

/*-----插入--------------*/
/**
 * 插入记录
 * @param Model 实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Insert = function ( Model , CallBack ) {
    var Sql = "Insert Into SFJ Set SFJ_State=?SFJ_State,SFJ_Name=?SFJ_Name,SFJ_CreateTime=?SFJ_CreateTime,SFJ_UpdateTime=?SFJ_UpdateTime,SFJ_ProdCount=?SFJ_ProdCount,SFJSiteLabelID=?SFJSiteLabelID,SFJLikeCount=?SFJLikeCount,SFJ_LabelID=?SFJ_LabelID,SFJ_UserID=?SFJ_UserID,SFJ_D_Income=?SFJ_D_Income,SFJ_TempStyle=?SFJ_TempStyle,SFJ_EvalCount=?SFJ_EvalCount,SFJ_ForWardCount=?SFJ_ForWardCount,SFJ_Users=?SFJ_Users,SFJ_Remark=?SFJ_Remark,SFJFailContent=?SFJFailContent ";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 插入记录
 * @param ModelList 实体集合 ，Type:Object, 实例：[{key1:value1,key2:value2},{key1:value1,key2:value2}]
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.InsertList = function ( ModelList , CallBack ) {
    var Sql = "Insert Into SFJ Set SFJ_State=?SFJ_State,SFJ_Name=?SFJ_Name,SFJ_CreateTime=?SFJ_CreateTime,SFJ_UpdateTime=?SFJ_UpdateTime,SFJ_ProdCount=?SFJ_ProdCount,SFJSiteLabelID=?SFJSiteLabelID,SFJLikeCount=?SFJLikeCount,SFJ_LabelID=?SFJ_LabelID,SFJ_UserID=?SFJ_UserID,SFJ_D_Income=?SFJ_D_Income,SFJ_TempStyle=?SFJ_TempStyle,SFJ_EvalCount=?SFJ_EvalCount,SFJ_ForWardCount=?SFJ_ForWardCount,SFJ_Users=?SFJ_Users,SFJ_Remark=?SFJ_Remark,SFJFailContent=?SFJFailContent";
    client.GetConnection ( function ( err , connection ) {
        if ( err ) {
            CallBack ( err , null );
            return;
        }
        for ( var i = 0 ; i < ModelList.length ; i ++ ) {
            var Model = ModelList[0];
            client.Query ( connection , Sql , Model , function ( err ) {
                if ( err ) {
                    event.emit ( 'InsertFail' );
                    return;
                }
                if ( i == ModelList.length - 1 ) {
                    event.emit ( 'InsertSuccess' );
                }
            } )
        }
        event.on ( 'InsertSuccess' , function () {
            connection.release ();
            CallBack ( true , null );
        } )
        event.on ( 'InsertFail' , function () {
            connection.release ();
            CallBack ( false , null );
        } )
    } )


}

/*-----更新--------------*/
/**
 * 更新记录
 * @param Model   实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Update = function ( Model , CallBack ) {
    var Sql = "Update SFJ Set  SFJ_State=?SFJ_State,SFJ_Name=?SFJ_Name,SFJ_CreateTime=?SFJ_CreateTime,SFJ_UpdateTime=?SFJ_UpdateTime,SFJ_ProdCount=?SFJ_ProdCount,SFJSiteLabelID=?SFJSiteLabelID,SFJLikeCount=?SFJLikeCount,SFJ_LabelID=?SFJ_LabelID,SFJ_UserID=?SFJ_UserID,SFJ_D_Income=?SFJ_D_Income,SFJ_TempStyle=?SFJ_TempStyle,SFJ_EvalCount=?SFJ_EvalCount,SFJ_ForWardCount=?SFJ_ForWardCount,SFJ_Users=?SFJ_Users,SFJ_Remark=?SFJ_Remark,SFJFailContent=?SFJFailContent Where SFJ_ID=?SFJ_ID";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 更新字段值 SFJ_State
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_State的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_State = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_State=?SFJ_State where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJ_Name
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_Name的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_Name = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_Name=?SFJ_Name where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJ_CreateTime
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_CreateTime的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_CreateTime = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_CreateTime=?SFJ_CreateTime where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJ_UpdateTime
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_UpdateTime的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_UpdateTime = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_UpdateTime=?SFJ_UpdateTime where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJ_ProdCount
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_ProdCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_ProdCount = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_ProdCount=?SFJ_ProdCount where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJSiteLabelID
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJSiteLabelID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJSiteLabelID = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJSiteLabelID=?SFJSiteLabelID where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJLikeCount
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJLikeCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJLikeCount = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJLikeCount=?SFJLikeCount where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJ_LabelID
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_LabelID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_LabelID = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_LabelID=?SFJ_LabelID where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJ_UserID
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_UserID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_UserID = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_UserID=?SFJ_UserID where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJ_D_Income
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_D_Income的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_D_Income = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_D_Income=?SFJ_D_Income where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJ_TempStyle
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_TempStyle的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_TempStyle = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_TempStyle=?SFJ_TempStyle where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJ_EvalCount
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_EvalCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_EvalCount = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_EvalCount=?SFJ_EvalCount where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJ_ForWardCount
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_ForWardCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_ForWardCount = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_ForWardCount=?SFJ_ForWardCount where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJ_Users
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_Users的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_Users = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_Users=?SFJ_Users where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJ_Remark
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJ_Remark的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJ_Remark = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJ_Remark=?SFJ_Remark where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SFJFailContent
 * @param Key  主键值 Key:SFJ_ID
 * @param Value SFJFailContent的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SFJFailContent = function ( SFJ_ID , Value , CallBack ) {
    var Sql = "Update SFJ Set SFJFailContent=?SFJFailContent where SFJ_ID=?SFJ_ID";
    var Model = {};
    Model.Value = Value;
    Model.SFJ_ID = SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/*-----删除数据--------------*/
/**
 *  根据主键删除数据
 * @param Key     主键值
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.Delete=function(SFJ_ID,CallBack){
    var Sql = "Delete From SFJ Where SFJ_ID=?SFJ_ID";
    var Model={};
    Model.SFJ_ID=SFJ_ID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *   删除所有数据
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.DeleteAll=function(CallBack){
    var Sql = "Delete From SFJ";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter=function(Filter,CallBack){
    var Sql = "Delete From SFJ where "+Filter+"";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter_Parms=function(Filter,Values,CallBack){
    var Sql = "Delete From SFJ where "+Filter+"";
    client.Query ( Sql  ,Values, function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *    删除所有记录
 *    重载方法说明：
 *   根据条件删除 - 非参数化
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   根据条件删除 - 参数化
 *   function(String 查询条件，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports. DeleteByFilter=function(){
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        DeleteByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Object, Function] , arguments ) ) {
        DeleteByFilter_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SFJ.js=>DeleteByFilter 缺少参数");
    CallBack(false,null);
}


/*-----获取分页数据--------------*/
var GetPageRow = function ( CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from SFJ  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field = function ( Field , CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from SFJ  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Parms = function ( CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from SFJ  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field_Parms = function ( Field , CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from SFJ  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


/**
 *    获取分页数据
 *    重载方法说明：
 *   获取所有字段
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *   获取指定字段
 *   function(String 要查询的字段（Filed1,Filed2）, Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *    获取所有字段  - 参数化查询
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ，Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ， 回调函数Function（err,Result）)
 *   获取指定字段 - 参数化查询
 *   function(String 要查询的字段（Filed1,Filed2）,  Number 当前页 ，Number 分页尺寸 , String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.GetPageRow=function(){
    if ( Tools.CheckArgs ( [Number , Number , String , Function] , arguments ) ) {
        GetPageRow ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String , Function] , arguments ) ) {
        GetPageRow_Field ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [ Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Field_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4],arguments[5] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SFJ.js=>GetPageRow 缺少参数");
    CallBack(false,null);
}
