// NewPromotion/Access/Shop.js
/**
 * Created with CodeModel.
 * User: 田林
 * Date: 2013-09-20
 * Time: 12:16:13
 * To change this template use File | Settings | File Templates.
 */
var client = require ( '../MySql' );
var Tools = require ( '../Tools' );
var EventEmitter = require ( 'events' ).EventEmitter;
var event = new EventEmitter ();

/*-----获取记录数--------------*/
var GetRsCount=function(Filter,CallBack){
    var Sql = "Select Count(*) as Count From Shop Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

var GetRsCount_Parms=function(Filter,Values ,CallBack){
    var Sql = "Select Count(*) as Count From Shop Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

/**
 * 功能:获取记录数
 *   重载方法说明：
 *   参数化查询
 *   function(String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *   非参数化查询
 *   function(String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.GetRsCount = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        GetRsCount ( arguments[0] ,arguments[1]);
        return;
    }
    if ( Tools.CheckArgs ( [String ,Object, Function] , arguments ) ) {
        GetRsCount_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/Shop.js=>GetRsCount 缺少参数");
    CallBack(false,null);
}
/*-----统计求和--------------*/
var Sum=function(Field,Filter,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From Shop Where "+Filter+"";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

var Sum_Parms=function(Field,Filter,Values ,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From Shop Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

/**
 * 功能：统计字段数值
 * 重载方法说明：
 * 参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *    非参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.Sum = function () {
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        Sum ( arguments[0] ,arguments[1],arguments[2]);
        return;
    }
    if ( Tools.CheckArgs ( [String, String ,Object, Function] , arguments ) ) {
        Sum_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/Shop.js=>Sum 缺少参数");
    CallBack(false,null);
}

/*-----查询所有记录--------------*/
var SelectAll = function ( CallBack ) {
    var Sql = "Select * From Shop";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectAll_Field = function (Field, CallBack ) {
    var Sql = "Select "+Field+" From Shop";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


var SelectAllByOrder = function (Field ,  Order , CallBack ) {
    var Sql = "Select "+Field+" From Shop  Order By " + Order;
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 功能: 查询所有记录
 *   重载方法说明：
 *   获取所有记录
 *   function(回调函数Function（err,Result）)
 *   获取所有记录 - 指定要返回的字段
 *   function(String 要返回的字段（Field1，Field2） ，回调函数Function（err,Result）)
 *   获取所有记录 - 排序
 *   function(String 要返回的字段（Field1，Field2） , String 排序表达式（Filed Desc），回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectAll = function () {
    if ( Tools.CheckArgs ( [Function] , arguments ) ) {
        SelectAll ( arguments[0] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        SelectAll_Field ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        SelectAllByOrder ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/Shop.js=>SelectAll 缺少参数");
    CallBack(false,null);
}
/*-----根据关键字查询记录--------------*/

var Select = function ( ShopID , CallBack ) {
    var Sql = "Select * From Shop Where ShopID=?ShopID";
    var Values = {};
    Values.ShopID = ShopID;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

    var Select_Field = function (Field, ShopID , CallBack ) {
    var Sql = "Select "+Field+" From Shop Where ShopID=?ShopID";
    var Values = {};
    Values.ShopID = ShopID;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

/**
 * 功能: 根据主键查询记录
 *   重载方法说明：
 *   返回所有字段  - 主键值为字符串
 *   function(String 主键值 ，回调函数Function（err,Result）)
 *   返回所有字段   - 主键值为数字
 *   function(Number 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段  - 主键值为字符串
 *   function(String 要返回的字段（Field1，Field2） ，String 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段 - 主键值为数字
 *   function(String 要返回的字段（Field1，Field2） ，Number 主键值 ，回调函数Function（err,Result）)
 * @constructor
 */
exports.Select = function () {

    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [Number, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Number, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/Shop.js=>Select 缺少参数");
    CallBack(false,null);
}

/*-----根据条件查询记录--------------*/
var SelectByFilter = function ( Filter , CallBack ) {
    var Sql = "Select * From Shop Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field = function (Field, Filter , CallBack ) {
    var Sql = "Select "+Field+" From Shop Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Parms = function ( Filter , Values , CallBack ) {
    var Sql = "Select * From Shop Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field_Parms = function (Field, Filter , Values , CallBack ) {
    var Sql = "Select "+Field+" From Shop Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 功能: 根据条件查询记录
 *   重载方法说明：
 *   返回所有字段 - 非参数化查询
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   返回指定字段 - 非参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ，回调函数Function（err,Result）)
 *   返回所有字段  - 参数化查询
 *   function(String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ， 回调函数Function（err,Result）)
 *   返回指定的字段 - 参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectByFilter = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        SelectByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        SelectByFilter_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Object , Function] , arguments ) ) {
        SelectByFilter_Parms ( arguments[0] , arguments[1] , arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Object , Function] , arguments ) ) {
        SelectByFilter_Field_Parms ( arguments[0] , arguments[1] , arguments[2],arguments[3] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/Shop.js=>SelectByFilter 缺少参数");
    CallBack(false,null);
}

/*-----插入--------------*/
/**
 * 插入记录
 * @param Model 实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Insert = function ( Model , CallBack ) {
    var Sql = "Insert Into Shop Set ShopID=?ShopID,ShopName=?ShopName,ShopNameSpell=?ShopNameSpell,ShopLabelID=?ShopLabelID,commission_Rate=?commission_Rate,PayPeriod=?PayPeriod,TagetUrl=?TagetUrl,LogoImg=?LogoImg,FanliUrl=?FanliUrl,ShopType=?ShopType,OrderCount=?OrderCount,ReadCount=?ReadCount,UserCount=?UserCount,RequestCount=?RequestCount,Visabled=?Visabled,SortCount=?SortCount,IsrEcommend=?IsrEcommend,ShopLabelInfo=?ShopLabelInfo,ShopInfo=?ShopInfo,FanliInfo=?FanliInfo,NoFanliInfo=?NoFanliInfo,AntherInfo=?AntherInfo,LastSellTime=?LastSellTime,HPCount=?HPCount,ZPCount=?ZPCount,CPCount=?CPCount,Shop_ForWardCount=?Shop_ForWardCount,ShopEvalCount=?ShopEvalCount ";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 插入记录
 * @param ModelList 实体集合 ，Type:Object, 实例：[{key1:value1,key2:value2},{key1:value1,key2:value2}]
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.InsertList = function ( ModelList , CallBack ) {
    var Sql = "Insert Into Shop Set ShopID=?ShopID,ShopName=?ShopName,ShopNameSpell=?ShopNameSpell,ShopLabelID=?ShopLabelID,commission_Rate=?commission_Rate,PayPeriod=?PayPeriod,TagetUrl=?TagetUrl,LogoImg=?LogoImg,FanliUrl=?FanliUrl,ShopType=?ShopType,OrderCount=?OrderCount,ReadCount=?ReadCount,UserCount=?UserCount,RequestCount=?RequestCount,Visabled=?Visabled,SortCount=?SortCount,IsrEcommend=?IsrEcommend,ShopLabelInfo=?ShopLabelInfo,ShopInfo=?ShopInfo,FanliInfo=?FanliInfo,NoFanliInfo=?NoFanliInfo,AntherInfo=?AntherInfo,LastSellTime=?LastSellTime,HPCount=?HPCount,ZPCount=?ZPCount,CPCount=?CPCount,Shop_ForWardCount=?Shop_ForWardCount,ShopEvalCount=?ShopEvalCount";
    client.GetConnection ( function ( err , connection ) {
        if ( err ) {
            CallBack ( err , null );
            return;
        }
        for ( var i = 0 ; i < ModelList.length ; i ++ ) {
            var Model = ModelList[0];
            client.Query ( connection , Sql , Model , function ( err ) {
                if ( err ) {
                    event.emit ( 'InsertFail' );
                    return;
                }
                if ( i == ModelList.length - 1 ) {
                    event.emit ( 'InsertSuccess' );
                }
            } )
        }
        event.on ( 'InsertSuccess' , function () {
            connection.release ();
            CallBack ( true , null );
        } )
        event.on ( 'InsertFail' , function () {
            connection.release ();
            CallBack ( false , null );
        } )
    } )


}

/*-----更新--------------*/
/**
 * 更新记录
 * @param Model   实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Update = function ( Model , CallBack ) {
    var Sql = "Update Shop Set  ShopID=?ShopID,ShopName=?ShopName,ShopNameSpell=?ShopNameSpell,ShopLabelID=?ShopLabelID,commission_Rate=?commission_Rate,PayPeriod=?PayPeriod,TagetUrl=?TagetUrl,LogoImg=?LogoImg,FanliUrl=?FanliUrl,ShopType=?ShopType,OrderCount=?OrderCount,ReadCount=?ReadCount,UserCount=?UserCount,RequestCount=?RequestCount,Visabled=?Visabled,SortCount=?SortCount,IsrEcommend=?IsrEcommend,ShopLabelInfo=?ShopLabelInfo,ShopInfo=?ShopInfo,FanliInfo=?FanliInfo,NoFanliInfo=?NoFanliInfo,AntherInfo=?AntherInfo,LastSellTime=?LastSellTime,HPCount=?HPCount,ZPCount=?ZPCount,CPCount=?CPCount,Shop_ForWardCount=?Shop_ForWardCount,ShopEvalCount=?ShopEvalCount Where ShopID=?ShopID";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 更新字段值 ShopID
 * @param Key  主键值 Key:ShopID
 * @param Value ShopID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ShopID = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set ShopID=?ShopID where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ShopName
 * @param Key  主键值 Key:ShopID
 * @param Value ShopName的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ShopName = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set ShopName=?ShopName where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ShopNameSpell
 * @param Key  主键值 Key:ShopID
 * @param Value ShopNameSpell的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ShopNameSpell = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set ShopNameSpell=?ShopNameSpell where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ShopLabelID
 * @param Key  主键值 Key:ShopID
 * @param Value ShopLabelID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ShopLabelID = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set ShopLabelID=?ShopLabelID where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 commission_Rate
 * @param Key  主键值 Key:ShopID
 * @param Value commission_Rate的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_commission_Rate = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set commission_Rate=?commission_Rate where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 PayPeriod
 * @param Key  主键值 Key:ShopID
 * @param Value PayPeriod的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_PayPeriod = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set PayPeriod=?PayPeriod where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 TagetUrl
 * @param Key  主键值 Key:ShopID
 * @param Value TagetUrl的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_TagetUrl = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set TagetUrl=?TagetUrl where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 LogoImg
 * @param Key  主键值 Key:ShopID
 * @param Value LogoImg的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_LogoImg = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set LogoImg=?LogoImg where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 FanliUrl
 * @param Key  主键值 Key:ShopID
 * @param Value FanliUrl的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_FanliUrl = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set FanliUrl=?FanliUrl where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ShopType
 * @param Key  主键值 Key:ShopID
 * @param Value ShopType的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ShopType = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set ShopType=?ShopType where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 OrderCount
 * @param Key  主键值 Key:ShopID
 * @param Value OrderCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_OrderCount = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set OrderCount=?OrderCount where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ReadCount
 * @param Key  主键值 Key:ShopID
 * @param Value ReadCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ReadCount = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set ReadCount=?ReadCount where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserCount
 * @param Key  主键值 Key:ShopID
 * @param Value UserCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserCount = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set UserCount=?UserCount where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 RequestCount
 * @param Key  主键值 Key:ShopID
 * @param Value RequestCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_RequestCount = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set RequestCount=?RequestCount where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 Visabled
 * @param Key  主键值 Key:ShopID
 * @param Value Visabled的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_Visabled = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set Visabled=?Visabled where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SortCount
 * @param Key  主键值 Key:ShopID
 * @param Value SortCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SortCount = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set SortCount=?SortCount where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 IsrEcommend
 * @param Key  主键值 Key:ShopID
 * @param Value IsrEcommend的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_IsrEcommend = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set IsrEcommend=?IsrEcommend where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ShopLabelInfo
 * @param Key  主键值 Key:ShopID
 * @param Value ShopLabelInfo的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ShopLabelInfo = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set ShopLabelInfo=?ShopLabelInfo where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ShopInfo
 * @param Key  主键值 Key:ShopID
 * @param Value ShopInfo的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ShopInfo = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set ShopInfo=?ShopInfo where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 FanliInfo
 * @param Key  主键值 Key:ShopID
 * @param Value FanliInfo的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_FanliInfo = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set FanliInfo=?FanliInfo where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 NoFanliInfo
 * @param Key  主键值 Key:ShopID
 * @param Value NoFanliInfo的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_NoFanliInfo = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set NoFanliInfo=?NoFanliInfo where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 AntherInfo
 * @param Key  主键值 Key:ShopID
 * @param Value AntherInfo的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_AntherInfo = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set AntherInfo=?AntherInfo where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 LastSellTime
 * @param Key  主键值 Key:ShopID
 * @param Value LastSellTime的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_LastSellTime = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set LastSellTime=?LastSellTime where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 HPCount
 * @param Key  主键值 Key:ShopID
 * @param Value HPCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_HPCount = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set HPCount=?HPCount where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ZPCount
 * @param Key  主键值 Key:ShopID
 * @param Value ZPCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ZPCount = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set ZPCount=?ZPCount where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 CPCount
 * @param Key  主键值 Key:ShopID
 * @param Value CPCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_CPCount = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set CPCount=?CPCount where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 Shop_ForWardCount
 * @param Key  主键值 Key:ShopID
 * @param Value Shop_ForWardCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_Shop_ForWardCount = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set Shop_ForWardCount=?Shop_ForWardCount where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ShopEvalCount
 * @param Key  主键值 Key:ShopID
 * @param Value ShopEvalCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ShopEvalCount = function ( ShopID , Value , CallBack ) {
    var Sql = "Update Shop Set ShopEvalCount=?ShopEvalCount where ShopID=?ShopID";
    var Model = {};
    Model.Value = Value;
    Model.ShopID = ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/*-----删除数据--------------*/
/**
 *  根据主键删除数据
 * @param Key     主键值
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.Delete=function(ShopID,CallBack){
    var Sql = "Delete From Shop Where ShopID=?ShopID";
    var Model={};
    Model.ShopID=ShopID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *   删除所有数据
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.DeleteAll=function(CallBack){
    var Sql = "Delete From Shop";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter=function(Filter,CallBack){
    var Sql = "Delete From Shop where "+Filter+"";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter_Parms=function(Filter,Values,CallBack){
    var Sql = "Delete From Shop where "+Filter+"";
    client.Query ( Sql  ,Values, function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *    删除所有记录
 *    重载方法说明：
 *   根据条件删除 - 非参数化
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   根据条件删除 - 参数化
 *   function(String 查询条件，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports. DeleteByFilter=function(){
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        DeleteByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Object, Function] , arguments ) ) {
        DeleteByFilter_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/Shop.js=>DeleteByFilter 缺少参数");
    CallBack(false,null);
}


/*-----获取分页数据--------------*/
var GetPageRow = function ( CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from Shop  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field = function ( Field , CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from Shop  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Parms = function ( CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from Shop  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field_Parms = function ( Field , CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from Shop  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


/**
 *    获取分页数据
 *    重载方法说明：
 *   获取所有字段
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *   获取指定字段
 *   function(String 要查询的字段（Filed1,Filed2）, Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *    获取所有字段  - 参数化查询
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ，Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ， 回调函数Function（err,Result）)
 *   获取指定字段 - 参数化查询
 *   function(String 要查询的字段（Filed1,Filed2）,  Number 当前页 ，Number 分页尺寸 , String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.GetPageRow=function(){
    if ( Tools.CheckArgs ( [Number , Number , String , Function] , arguments ) ) {
        GetPageRow ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String , Function] , arguments ) ) {
        GetPageRow_Field ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [ Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Field_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4],arguments[5] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/Shop.js=>GetPageRow 缺少参数");
    CallBack(false,null);
}
