// NewPromotion/Access/SiteProd.js
/**
 * Created with CodeModel.
 * User: 田林
 * Date: 2013-09-20
 * Time: 12:16:13
 * To change this template use File | Settings | File Templates.
 */
var client = require ( '../MySql' );
var Tools = require ( '../Tools' );
var EventEmitter = require ( 'events' ).EventEmitter;
var event = new EventEmitter ();

/*-----获取记录数--------------*/
var GetRsCount=function(Filter,CallBack){
    var Sql = "Select Count(*) as Count From SiteProd Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

var GetRsCount_Parms=function(Filter,Values ,CallBack){
    var Sql = "Select Count(*) as Count From SiteProd Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

/**
 * 功能:获取记录数
 *   重载方法说明：
 *   参数化查询
 *   function(String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *   非参数化查询
 *   function(String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.GetRsCount = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        GetRsCount ( arguments[0] ,arguments[1]);
        return;
    }
    if ( Tools.CheckArgs ( [String ,Object, Function] , arguments ) ) {
        GetRsCount_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteProd.js=>GetRsCount 缺少参数");
    CallBack(false,null);
}
/*-----统计求和--------------*/
var Sum=function(Field,Filter,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From SiteProd Where "+Filter+"";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

var Sum_Parms=function(Field,Filter,Values ,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From SiteProd Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

/**
 * 功能：统计字段数值
 * 重载方法说明：
 * 参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *    非参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.Sum = function () {
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        Sum ( arguments[0] ,arguments[1],arguments[2]);
        return;
    }
    if ( Tools.CheckArgs ( [String, String ,Object, Function] , arguments ) ) {
        Sum_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteProd.js=>Sum 缺少参数");
    CallBack(false,null);
}

/*-----查询所有记录--------------*/
var SelectAll = function ( CallBack ) {
    var Sql = "Select * From SiteProd";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectAll_Field = function (Field, CallBack ) {
    var Sql = "Select "+Field+" From SiteProd";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


var SelectAllByOrder = function (Field ,  Order , CallBack ) {
    var Sql = "Select "+Field+" From SiteProd  Order By " + Order;
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 功能: 查询所有记录
 *   重载方法说明：
 *   获取所有记录
 *   function(回调函数Function（err,Result）)
 *   获取所有记录 - 指定要返回的字段
 *   function(String 要返回的字段（Field1，Field2） ，回调函数Function（err,Result）)
 *   获取所有记录 - 排序
 *   function(String 要返回的字段（Field1，Field2） , String 排序表达式（Filed Desc），回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectAll = function () {
    if ( Tools.CheckArgs ( [Function] , arguments ) ) {
        SelectAll ( arguments[0] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        SelectAll_Field ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        SelectAllByOrder ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteProd.js=>SelectAll 缺少参数");
    CallBack(false,null);
}
/*-----根据关键字查询记录--------------*/

var Select = function ( ProdID , CallBack ) {
    var Sql = "Select * From SiteProd Where ProdID=?ProdID";
    var Values = {};
    Values.ProdID = ProdID;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

    var Select_Field = function (Field, ProdID , CallBack ) {
    var Sql = "Select "+Field+" From SiteProd Where ProdID=?ProdID";
    var Values = {};
    Values.ProdID = ProdID;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

/**
 * 功能: 根据主键查询记录
 *   重载方法说明：
 *   返回所有字段  - 主键值为字符串
 *   function(String 主键值 ，回调函数Function（err,Result）)
 *   返回所有字段   - 主键值为数字
 *   function(Number 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段  - 主键值为字符串
 *   function(String 要返回的字段（Field1，Field2） ，String 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段 - 主键值为数字
 *   function(String 要返回的字段（Field1，Field2） ，Number 主键值 ，回调函数Function（err,Result）)
 * @constructor
 */
exports.Select = function () {

    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [Number, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Number, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteProd.js=>Select 缺少参数");
    CallBack(false,null);
}

/*-----根据条件查询记录--------------*/
var SelectByFilter = function ( Filter , CallBack ) {
    var Sql = "Select * From SiteProd Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field = function (Field, Filter , CallBack ) {
    var Sql = "Select "+Field+" From SiteProd Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Parms = function ( Filter , Values , CallBack ) {
    var Sql = "Select * From SiteProd Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field_Parms = function (Field, Filter , Values , CallBack ) {
    var Sql = "Select "+Field+" From SiteProd Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 功能: 根据条件查询记录
 *   重载方法说明：
 *   返回所有字段 - 非参数化查询
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   返回指定字段 - 非参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ，回调函数Function（err,Result）)
 *   返回所有字段  - 参数化查询
 *   function(String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ， 回调函数Function（err,Result）)
 *   返回指定的字段 - 参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectByFilter = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        SelectByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        SelectByFilter_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Object , Function] , arguments ) ) {
        SelectByFilter_Parms ( arguments[0] , arguments[1] , arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Object , Function] , arguments ) ) {
        SelectByFilter_Field_Parms ( arguments[0] , arguments[1] , arguments[2],arguments[3] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteProd.js=>SelectByFilter 缺少参数");
    CallBack(false,null);
}

/*-----插入--------------*/
/**
 * 插入记录
 * @param Model 实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Insert = function ( Model , CallBack ) {
    var Sql = "Insert Into SiteProd Set ProdID=?ProdID,SellerNick=?SellerNick,ProdSiteLbel=?ProdSiteLbel,ProdName=?ProdName,ProdPrice=?ProdPrice,ProdPromotionPrice=?ProdPromotionPrice,IsBagmail=?IsBagmail,commission=?commission,commission_rate=?commission_rate,pic_url=?pic_url,Wtime=?Wtime,HotCount=?HotCount,LikeCount=?LikeCount,ShopType=?ShopType,Color=?Color,XMCount=?XMCount,MGCount=?MGCount,XMPoint=?XMPoint,MGPoint=?MGPoint,EvalCount=?EvalCount,ShareCount=?ShareCount,XYCount=?XYCount,QYJCount=?QYJCount,ForWardCount=?ForWardCount,IsNew=?IsNew,IsOutOfDate=?IsOutOfDate ";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 插入记录
 * @param ModelList 实体集合 ，Type:Object, 实例：[{key1:value1,key2:value2},{key1:value1,key2:value2}]
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.InsertList = function ( ModelList , CallBack ) {
    var Sql = "Insert Into SiteProd Set ProdID=?ProdID,SellerNick=?SellerNick,ProdSiteLbel=?ProdSiteLbel,ProdName=?ProdName,ProdPrice=?ProdPrice,ProdPromotionPrice=?ProdPromotionPrice,IsBagmail=?IsBagmail,commission=?commission,commission_rate=?commission_rate,pic_url=?pic_url,Wtime=?Wtime,HotCount=?HotCount,LikeCount=?LikeCount,ShopType=?ShopType,Color=?Color,XMCount=?XMCount,MGCount=?MGCount,XMPoint=?XMPoint,MGPoint=?MGPoint,EvalCount=?EvalCount,ShareCount=?ShareCount,XYCount=?XYCount,QYJCount=?QYJCount,ForWardCount=?ForWardCount,IsNew=?IsNew,IsOutOfDate=?IsOutOfDate";
    client.GetConnection ( function ( err , connection ) {
        if ( err ) {
            CallBack ( err , null );
            return;
        }
        for ( var i = 0 ; i < ModelList.length ; i ++ ) {
            var Model = ModelList[0];
            client.Query ( connection , Sql , Model , function ( err ) {
                if ( err ) {
                    event.emit ( 'InsertFail' );
                    return;
                }
                if ( i == ModelList.length - 1 ) {
                    event.emit ( 'InsertSuccess' );
                }
            } )
        }
        event.on ( 'InsertSuccess' , function () {
            connection.release ();
            CallBack ( true , null );
        } )
        event.on ( 'InsertFail' , function () {
            connection.release ();
            CallBack ( false , null );
        } )
    } )


}

/*-----更新--------------*/
/**
 * 更新记录
 * @param Model   实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Update = function ( Model , CallBack ) {
    var Sql = "Update SiteProd Set  ProdID=?ProdID,SellerNick=?SellerNick,ProdSiteLbel=?ProdSiteLbel,ProdName=?ProdName,ProdPrice=?ProdPrice,ProdPromotionPrice=?ProdPromotionPrice,IsBagmail=?IsBagmail,commission=?commission,commission_rate=?commission_rate,pic_url=?pic_url,Wtime=?Wtime,HotCount=?HotCount,LikeCount=?LikeCount,ShopType=?ShopType,Color=?Color,XMCount=?XMCount,MGCount=?MGCount,XMPoint=?XMPoint,MGPoint=?MGPoint,EvalCount=?EvalCount,ShareCount=?ShareCount,XYCount=?XYCount,QYJCount=?QYJCount,ForWardCount=?ForWardCount,IsNew=?IsNew,IsOutOfDate=?IsOutOfDate Where ProdID=?ProdID";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 更新字段值 ProdID
 * @param Key  主键值 Key:ProdID
 * @param Value ProdID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ProdID = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set ProdID=?ProdID where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 SellerNick
 * @param Key  主键值 Key:ProdID
 * @param Value SellerNick的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_SellerNick = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set SellerNick=?SellerNick where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ProdSiteLbel
 * @param Key  主键值 Key:ProdID
 * @param Value ProdSiteLbel的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ProdSiteLbel = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set ProdSiteLbel=?ProdSiteLbel where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ProdName
 * @param Key  主键值 Key:ProdID
 * @param Value ProdName的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ProdName = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set ProdName=?ProdName where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ProdPrice
 * @param Key  主键值 Key:ProdID
 * @param Value ProdPrice的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ProdPrice = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set ProdPrice=?ProdPrice where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ProdPromotionPrice
 * @param Key  主键值 Key:ProdID
 * @param Value ProdPromotionPrice的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ProdPromotionPrice = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set ProdPromotionPrice=?ProdPromotionPrice where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 IsBagmail
 * @param Key  主键值 Key:ProdID
 * @param Value IsBagmail的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_IsBagmail = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set IsBagmail=?IsBagmail where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 commission
 * @param Key  主键值 Key:ProdID
 * @param Value commission的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_commission = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set commission=?commission where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 commission_rate
 * @param Key  主键值 Key:ProdID
 * @param Value commission_rate的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_commission_rate = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set commission_rate=?commission_rate where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 pic_url
 * @param Key  主键值 Key:ProdID
 * @param Value pic_url的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_pic_url = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set pic_url=?pic_url where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 Wtime
 * @param Key  主键值 Key:ProdID
 * @param Value Wtime的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_Wtime = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set Wtime=?Wtime where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 HotCount
 * @param Key  主键值 Key:ProdID
 * @param Value HotCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_HotCount = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set HotCount=?HotCount where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 LikeCount
 * @param Key  主键值 Key:ProdID
 * @param Value LikeCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_LikeCount = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set LikeCount=?LikeCount where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ShopType
 * @param Key  主键值 Key:ProdID
 * @param Value ShopType的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ShopType = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set ShopType=?ShopType where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 Color
 * @param Key  主键值 Key:ProdID
 * @param Value Color的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_Color = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set Color=?Color where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 XMCount
 * @param Key  主键值 Key:ProdID
 * @param Value XMCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_XMCount = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set XMCount=?XMCount where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 MGCount
 * @param Key  主键值 Key:ProdID
 * @param Value MGCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_MGCount = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set MGCount=?MGCount where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 XMPoint
 * @param Key  主键值 Key:ProdID
 * @param Value XMPoint的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_XMPoint = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set XMPoint=?XMPoint where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 MGPoint
 * @param Key  主键值 Key:ProdID
 * @param Value MGPoint的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_MGPoint = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set MGPoint=?MGPoint where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 EvalCount
 * @param Key  主键值 Key:ProdID
 * @param Value EvalCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_EvalCount = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set EvalCount=?EvalCount where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ShareCount
 * @param Key  主键值 Key:ProdID
 * @param Value ShareCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ShareCount = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set ShareCount=?ShareCount where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 XYCount
 * @param Key  主键值 Key:ProdID
 * @param Value XYCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_XYCount = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set XYCount=?XYCount where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 QYJCount
 * @param Key  主键值 Key:ProdID
 * @param Value QYJCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_QYJCount = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set QYJCount=?QYJCount where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ForWardCount
 * @param Key  主键值 Key:ProdID
 * @param Value ForWardCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ForWardCount = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set ForWardCount=?ForWardCount where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 IsNew
 * @param Key  主键值 Key:ProdID
 * @param Value IsNew的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_IsNew = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set IsNew=?IsNew where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 IsOutOfDate
 * @param Key  主键值 Key:ProdID
 * @param Value IsOutOfDate的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_IsOutOfDate = function ( ProdID , Value , CallBack ) {
    var Sql = "Update SiteProd Set IsOutOfDate=?IsOutOfDate where ProdID=?ProdID";
    var Model = {};
    Model.Value = Value;
    Model.ProdID = ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/*-----删除数据--------------*/
/**
 *  根据主键删除数据
 * @param Key     主键值
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.Delete=function(ProdID,CallBack){
    var Sql = "Delete From SiteProd Where ProdID=?ProdID";
    var Model={};
    Model.ProdID=ProdID;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *   删除所有数据
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.DeleteAll=function(CallBack){
    var Sql = "Delete From SiteProd";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter=function(Filter,CallBack){
    var Sql = "Delete From SiteProd where "+Filter+"";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter_Parms=function(Filter,Values,CallBack){
    var Sql = "Delete From SiteProd where "+Filter+"";
    client.Query ( Sql  ,Values, function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *    删除所有记录
 *    重载方法说明：
 *   根据条件删除 - 非参数化
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   根据条件删除 - 参数化
 *   function(String 查询条件，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports. DeleteByFilter=function(){
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        DeleteByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Object, Function] , arguments ) ) {
        DeleteByFilter_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteProd.js=>DeleteByFilter 缺少参数");
    CallBack(false,null);
}


/*-----获取分页数据--------------*/
var GetPageRow = function ( CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from SiteProd  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field = function ( Field , CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from SiteProd  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Parms = function ( CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from SiteProd  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field_Parms = function ( Field , CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from SiteProd  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


/**
 *    获取分页数据
 *    重载方法说明：
 *   获取所有字段
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *   获取指定字段
 *   function(String 要查询的字段（Filed1,Filed2）, Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *    获取所有字段  - 参数化查询
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ，Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ， 回调函数Function（err,Result）)
 *   获取指定字段 - 参数化查询
 *   function(String 要查询的字段（Filed1,Filed2）,  Number 当前页 ，Number 分页尺寸 , String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.GetPageRow=function(){
    if ( Tools.CheckArgs ( [Number , Number , String , Function] , arguments ) ) {
        GetPageRow ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String , Function] , arguments ) ) {
        GetPageRow_Field ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [ Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Field_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4],arguments[5] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteProd.js=>GetPageRow 缺少参数");
    CallBack(false,null);
}
