// NewPromotion/Access/SiteUser.js
/**
 * Created with CodeModel.
 * User: 田林
 * Date: 2013-09-20
 * Time: 12:16:13
 * To change this template use File | Settings | File Templates.
 */
var client = require ( '../MySql' );
var Tools = require ( '../Tools' );
var EventEmitter = require ( 'events' ).EventEmitter;
var event = new EventEmitter ();

/*-----获取记录数--------------*/
var GetRsCount=function(Filter,CallBack){
    var Sql = "Select Count(*) as Count From SiteUser Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

var GetRsCount_Parms=function(Filter,Values ,CallBack){
    var Sql = "Select Count(*) as Count From SiteUser Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].Count );
    } )
}

/**
 * 功能:获取记录数
 *   重载方法说明：
 *   参数化查询
 *   function(String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *   非参数化查询
 *   function(String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.GetRsCount = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        GetRsCount ( arguments[0] ,arguments[1]);
        return;
    }
    if ( Tools.CheckArgs ( [String ,Object, Function] , arguments ) ) {
        GetRsCount_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteUser.js=>GetRsCount 缺少参数");
    CallBack(false,null);
}
/*-----统计求和--------------*/
var Sum=function(Field,Filter,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From SiteUser Where "+Filter+"";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

var Sum_Parms=function(Field,Filter,Values ,CallBack){
    var Sql = "Select Sum("+Field+") as SumCount From SiteUser Where "+Filter+"";
    client.Query ( Sql ,Values, function ( err , Result ) {
        CallBack ( err , Result[0].SumCount );
    } )
}

/**
 * 功能：统计字段数值
 * 重载方法说明：
 * 参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ,回调函数Function（err,Result）)
 *    非参数化查询
 *   function(String 要统计的字段(字段名称),String 查询条件,回调函数Function（err,Result）)
 * @constructor
 */
exports.Sum = function () {
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        Sum ( arguments[0] ,arguments[1],arguments[2]);
        return;
    }
    if ( Tools.CheckArgs ( [String, String ,Object, Function] , arguments ) ) {
        Sum_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }

    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteUser.js=>Sum 缺少参数");
    CallBack(false,null);
}

/*-----查询所有记录--------------*/
var SelectAll = function ( CallBack ) {
    var Sql = "Select * From SiteUser";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectAll_Field = function (Field, CallBack ) {
    var Sql = "Select "+Field+" From SiteUser";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


var SelectAllByOrder = function (Field ,  Order , CallBack ) {
    var Sql = "Select "+Field+" From SiteUser  Order By " + Order;
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 功能: 查询所有记录
 *   重载方法说明：
 *   获取所有记录
 *   function(回调函数Function（err,Result）)
 *   获取所有记录 - 指定要返回的字段
 *   function(String 要返回的字段（Field1，Field2） ，回调函数Function（err,Result）)
 *   获取所有记录 - 排序
 *   function(String 要返回的字段（Field1，Field2） , String 排序表达式（Filed Desc），回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectAll = function () {
    if ( Tools.CheckArgs ( [Function] , arguments ) ) {
        SelectAll ( arguments[0] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        SelectAll_Field ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        SelectAllByOrder ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteUser.js=>SelectAll 缺少参数");
    CallBack(false,null);
}
/*-----根据关键字查询记录--------------*/

var Select = function ( UserId , CallBack ) {
    var Sql = "Select * From SiteUser Where UserId=?UserId";
    var Values = {};
    Values.UserId = UserId;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

    var Select_Field = function (Field, UserId , CallBack ) {
    var Sql = "Select "+Field+" From SiteUser Where UserId=?UserId";
    var Values = {};
    Values.UserId = UserId;
    client.Query ( Sql , Values , function ( err , Result ) {
        if ( Result.length > 0 ) {
            CallBack ( err , Result[0] );
        } else {
            CallBack ( err , null );
        }
    } )
}

/**
 * 功能: 根据主键查询记录
 *   重载方法说明：
 *   返回所有字段  - 主键值为字符串
 *   function(String 主键值 ，回调函数Function（err,Result）)
 *   返回所有字段   - 主键值为数字
 *   function(Number 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段  - 主键值为字符串
 *   function(String 要返回的字段（Field1，Field2） ，String 主键值 ，回调函数Function（err,Result）)
 *   返回指定的字段 - 主键值为数字
 *   function(String 要返回的字段（Field1，Field2） ，Number 主键值 ，回调函数Function（err,Result）)
 * @constructor
 */
exports.Select = function () {

    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [Number, Function] , arguments ) ) {
        Select ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Number, Function] , arguments ) ) {
        Select_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteUser.js=>Select 缺少参数");
    CallBack(false,null);
}

/*-----根据条件查询记录--------------*/
var SelectByFilter = function ( Filter , CallBack ) {
    var Sql = "Select * From SiteUser Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field = function (Field, Filter , CallBack ) {
    var Sql = "Select "+Field+" From SiteUser Where " + Filter + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Parms = function ( Filter , Values , CallBack ) {
    var Sql = "Select * From SiteUser Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var SelectByFilter_Field_Parms = function (Field, Filter , Values , CallBack ) {
    var Sql = "Select "+Field+" From SiteUser Where " + Filter + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 功能: 根据条件查询记录
 *   重载方法说明：
 *   返回所有字段 - 非参数化查询
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   返回指定字段 - 非参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ，回调函数Function（err,Result）)
 *   返回所有字段  - 参数化查询
 *   function(String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2}) ， 回调函数Function（err,Result）)
 *   返回指定的字段 - 参数化查询
 *   function(String 要返回的字段（Field1，Field2） ，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.SelectByFilter = function () {
    if ( Tools.CheckArgs ( [String , Function] , arguments ) ) {
        SelectByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String , Function] , arguments ) ) {
        SelectByFilter_Field ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String, Object , Function] , arguments ) ) {
        SelectByFilter_Parms ( arguments[0] , arguments[1] , arguments[2] );
        return;
    }
    if ( Tools.CheckArgs ( [String,String, Object , Function] , arguments ) ) {
        SelectByFilter_Field_Parms ( arguments[0] , arguments[1] , arguments[2],arguments[3] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteUser.js=>SelectByFilter 缺少参数");
    CallBack(false,null);
}

/*-----插入--------------*/
/**
 * 插入记录
 * @param Model 实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Insert = function ( Model , CallBack ) {
    var Sql = "Insert Into SiteUser Set UserEmail=?UserEmail,UserPwd=?UserPwd,IsActivate=?IsActivate,UserGender=?UserGender,UserRealName=?UserRealName,UserPingRealName=?UserPingRealName,UserCreatTime=?UserCreatTime,UserLoginTime=?UserLoginTime,UserCurMoney=?UserCurMoney,UserNeedPayMoney=?UserNeedPayMoney,UserPayMoney=?UserPayMoney,UserAllMoney=?UserAllMoney,arrears=?arrears,UserLev=?UserLev,TaobaoUserID=?TaobaoUserID,TaoBaoNick=?TaoBaoNick,ParentUserID=?ParentUserID,MyRelationCount=?MyRelationCount,RelationMeCount=?RelationMeCount,HeaderImg=?HeaderImg,UserSycWeiboLabel=?UserSycWeiboLabel ";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 插入记录
 * @param ModelList 实体集合 ，Type:Object, 实例：[{key1:value1,key2:value2},{key1:value1,key2:value2}]
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.InsertList = function ( ModelList , CallBack ) {
    var Sql = "Insert Into SiteUser Set UserEmail=?UserEmail,UserPwd=?UserPwd,IsActivate=?IsActivate,UserGender=?UserGender,UserRealName=?UserRealName,UserPingRealName=?UserPingRealName,UserCreatTime=?UserCreatTime,UserLoginTime=?UserLoginTime,UserCurMoney=?UserCurMoney,UserNeedPayMoney=?UserNeedPayMoney,UserPayMoney=?UserPayMoney,UserAllMoney=?UserAllMoney,arrears=?arrears,UserLev=?UserLev,TaobaoUserID=?TaobaoUserID,TaoBaoNick=?TaoBaoNick,ParentUserID=?ParentUserID,MyRelationCount=?MyRelationCount,RelationMeCount=?RelationMeCount,HeaderImg=?HeaderImg,UserSycWeiboLabel=?UserSycWeiboLabel";
    client.GetConnection ( function ( err , connection ) {
        if ( err ) {
            CallBack ( err , null );
            return;
        }
        for ( var i = 0 ; i < ModelList.length ; i ++ ) {
            var Model = ModelList[0];
            client.Query ( connection , Sql , Model , function ( err ) {
                if ( err ) {
                    event.emit ( 'InsertFail' );
                    return;
                }
                if ( i == ModelList.length - 1 ) {
                    event.emit ( 'InsertSuccess' );
                }
            } )
        }
        event.on ( 'InsertSuccess' , function () {
            connection.release ();
            CallBack ( true , null );
        } )
        event.on ( 'InsertFail' , function () {
            connection.release ();
            CallBack ( false , null );
        } )
    } )


}

/*-----更新--------------*/
/**
 * 更新记录
 * @param Model   实体 Type:Object, 实例：{key1:value1,key2:value2}
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Update = function ( Model , CallBack ) {
    var Sql = "Update SiteUser Set  UserEmail=?UserEmail,UserPwd=?UserPwd,IsActivate=?IsActivate,UserGender=?UserGender,UserRealName=?UserRealName,UserPingRealName=?UserPingRealName,UserCreatTime=?UserCreatTime,UserLoginTime=?UserLoginTime,UserCurMoney=?UserCurMoney,UserNeedPayMoney=?UserNeedPayMoney,UserPayMoney=?UserPayMoney,UserAllMoney=?UserAllMoney,arrears=?arrears,UserLev=?UserLev,TaobaoUserID=?TaobaoUserID,TaoBaoNick=?TaoBaoNick,ParentUserID=?ParentUserID,MyRelationCount=?MyRelationCount,RelationMeCount=?RelationMeCount,HeaderImg=?HeaderImg,UserSycWeiboLabel=?UserSycWeiboLabel Where UserId=?UserId";
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/**
 * 更新字段值 UserEmail
 * @param Key  主键值 Key:UserId
 * @param Value UserEmail的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserEmail = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserEmail=?UserEmail where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserPwd
 * @param Key  主键值 Key:UserId
 * @param Value UserPwd的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserPwd = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserPwd=?UserPwd where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 IsActivate
 * @param Key  主键值 Key:UserId
 * @param Value IsActivate的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_IsActivate = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set IsActivate=?IsActivate where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserGender
 * @param Key  主键值 Key:UserId
 * @param Value UserGender的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserGender = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserGender=?UserGender where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserRealName
 * @param Key  主键值 Key:UserId
 * @param Value UserRealName的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserRealName = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserRealName=?UserRealName where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserPingRealName
 * @param Key  主键值 Key:UserId
 * @param Value UserPingRealName的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserPingRealName = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserPingRealName=?UserPingRealName where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserCreatTime
 * @param Key  主键值 Key:UserId
 * @param Value UserCreatTime的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserCreatTime = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserCreatTime=?UserCreatTime where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserLoginTime
 * @param Key  主键值 Key:UserId
 * @param Value UserLoginTime的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserLoginTime = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserLoginTime=?UserLoginTime where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserCurMoney
 * @param Key  主键值 Key:UserId
 * @param Value UserCurMoney的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserCurMoney = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserCurMoney=?UserCurMoney where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserNeedPayMoney
 * @param Key  主键值 Key:UserId
 * @param Value UserNeedPayMoney的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserNeedPayMoney = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserNeedPayMoney=?UserNeedPayMoney where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserPayMoney
 * @param Key  主键值 Key:UserId
 * @param Value UserPayMoney的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserPayMoney = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserPayMoney=?UserPayMoney where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserAllMoney
 * @param Key  主键值 Key:UserId
 * @param Value UserAllMoney的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserAllMoney = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserAllMoney=?UserAllMoney where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 arrears
 * @param Key  主键值 Key:UserId
 * @param Value arrears的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_arrears = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set arrears=?arrears where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserLev
 * @param Key  主键值 Key:UserId
 * @param Value UserLev的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserLev = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserLev=?UserLev where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 TaobaoUserID
 * @param Key  主键值 Key:UserId
 * @param Value TaobaoUserID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_TaobaoUserID = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set TaobaoUserID=?TaobaoUserID where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 TaoBaoNick
 * @param Key  主键值 Key:UserId
 * @param Value TaoBaoNick的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_TaoBaoNick = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set TaoBaoNick=?TaoBaoNick where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 ParentUserID
 * @param Key  主键值 Key:UserId
 * @param Value ParentUserID的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_ParentUserID = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set ParentUserID=?ParentUserID where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 MyRelationCount
 * @param Key  主键值 Key:UserId
 * @param Value MyRelationCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_MyRelationCount = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set MyRelationCount=?MyRelationCount where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 RelationMeCount
 * @param Key  主键值 Key:UserId
 * @param Value RelationMeCount的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_RelationMeCount = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set RelationMeCount=?RelationMeCount where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 HeaderImg
 * @param Key  主键值 Key:UserId
 * @param Value HeaderImg的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_HeaderImg = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set HeaderImg=?HeaderImg where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 * 更新字段值 UserSycWeiboLabel
 * @param Key  主键值 Key:UserId
 * @param Value UserSycWeiboLabel的值
 * @param CallBack   回调函数Function（err,Result）
 * @constructor
 */
exports.Set_UserSycWeiboLabel = function ( UserId , Value , CallBack ) {
    var Sql = "Update SiteUser Set UserSycWeiboLabel=?UserSycWeiboLabel where UserId=?UserId";
    var Model = {};
    Model.Value = Value;
    Model.UserId = UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}



/*-----删除数据--------------*/
/**
 *  根据主键删除数据
 * @param Key     主键值
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.Delete=function(UserId,CallBack){
    var Sql = "Delete From SiteUser Where UserId=?UserId";
    var Model={};
    Model.UserId=UserId;
    client.Query ( Sql , Model , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *   删除所有数据
 * @param CallBack  回调函数Function（err,Result）
 * @constructor
 */
exports.DeleteAll=function(CallBack){
    var Sql = "Delete From SiteUser";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter=function(Filter,CallBack){
    var Sql = "Delete From SiteUser where "+Filter+"";
    client.Query ( Sql  , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var DeleteByFilter_Parms=function(Filter,Values,CallBack){
    var Sql = "Delete From SiteUser where "+Filter+"";
    client.Query ( Sql  ,Values, function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

/**
 *    删除所有记录
 *    重载方法说明：
 *   根据条件删除 - 非参数化
 *   function(String 查询条件 ，回调函数Function（err,Result）)
 *   根据条件删除 - 参数化
 *   function(String 查询条件，String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports. DeleteByFilter=function(){
    if ( Tools.CheckArgs ( [String, Function] , arguments ) ) {
        DeleteByFilter ( arguments[0] , arguments[1] );
        return;
    }
    if ( Tools.CheckArgs ( [String,Object, Function] , arguments ) ) {
        DeleteByFilter_Parms ( arguments[0] , arguments[1],arguments[2] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteUser.js=>DeleteByFilter 缺少参数");
    CallBack(false,null);
}


/*-----获取分页数据--------------*/
var GetPageRow = function ( CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from SiteUser  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field = function ( Field , CurPage , PageSize , Filter , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from SiteUser  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Parms = function ( CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select *  from SiteUser  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}

var GetPageRow_Field_Parms = function ( Field , CurPage , PageSize , Filter , Values , CallBack ) {
    if(Filter=="")  Filter="1=1";
    var Sql = "Select " + Field + "  from SiteUser  where " + Filter + "  limit " + (PageSize * (CurPage - 1)) + "," + PageSize + "";
    client.Query ( Sql , Values , function ( err , Result ) {
        CallBack ( err , Result );
    } )
}


/**
 *    获取分页数据
 *    重载方法说明：
 *   获取所有字段
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *   获取指定字段
 *   function(String 要查询的字段（Filed1,Filed2）, Number 当前页 ，Number 分页尺寸 , String  查询条件 ， 回调函数Function（err,Result）)
 *    获取所有字段  - 参数化查询
 *   function(Number 当前页 ，Number 分页尺寸 , String  查询条件 ，Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ， 回调函数Function（err,Result）)
 *   获取指定字段 - 参数化查询
 *   function(String 要查询的字段（Filed1,Filed2）,  Number 当前页 ，Number 分页尺寸 , String 查询条件 ， Object 参数化查询时的参数值的集合({key1:value1,key2:value2})  ，回调函数Function（err,Result）)
 * @constructor
 */
exports.GetPageRow=function(){
    if ( Tools.CheckArgs ( [Number , Number , String , Function] , arguments ) ) {
        GetPageRow ( arguments[0] , arguments[1],arguments[2],arguments[3] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String , Function] , arguments ) ) {
        GetPageRow_Field ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [ Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4] );
        return;
    }
    if ( Tools.CheckArgs ( [String , Number , Number , String ,Object, Function] , arguments ) ) {
        GetPageRow_Field_Parms ( arguments[0] , arguments[1],arguments[2],arguments[3],arguments[4],arguments[5] );
        return;
    }
    var CallBack= arguments[arguments.length-1];
    console.log("Access/SiteUser.js=>GetPageRow 缺少参数");
    CallBack(false,null);
}
