// NewPromotion/index.js
/**
 * Created with CodeModel.
 * User: 田林
 * Date: 2013-09-20
 * Time: 12:16:12
 * To change this template use File | Settings | File Templates.
 */
exports.Client = require('../MySql');

exports.App = require('./Logic/App');

exports.Authority = require('./Logic/Authority');

exports.BackGroundImg = require('./Logic/BackGroundImg');

exports.ClubGroup = require('./Logic/ClubGroup');

exports.ClubLabel = require('./Logic/ClubLabel');

exports.ClubModule = require('./Logic/ClubModule');

exports.ClubModuleAdmin = require('./Logic/ClubModuleAdmin');

exports.ClubReqThread = require('./Logic/ClubReqThread');

exports.ClubThread = require('./Logic/ClubThread');

exports.ClubThreadAgree = require('./Logic/ClubThreadAgree');

exports.ClubThreadSupport = require('./Logic/ClubThreadSupport');

exports.KeyWord = require('./Logic/KeyWord');

exports.KeyWordLabel = require('./Logic/KeyWordLabel');

exports.KeyWord_Prod = require('./Logic/KeyWord_Prod');

exports.OAuth = require('./Logic/OAuth');

exports.OrderCode = require('./Logic/OrderCode');

exports.OrderCode_User = require('./Logic/OrderCode_User');

exports.ProdEvaluate = require('./Logic/ProdEvaluate');

exports.Prod_User_MG = require('./Logic/Prod_User_MG');

exports.Prod_User_QYJ = require('./Logic/Prod_User_QYJ');

exports.Prod_User_Share = require('./Logic/Prod_User_Share');

exports.Prod_User_XM = require('./Logic/Prod_User_XM');

exports.Prod_User_XY = require('./Logic/Prod_User_XY');

exports.R_Group_User = require('./Logic/R_Group_User');

exports.R_Msg_User = require('./Logic/R_Msg_User');

exports.Role = require('./Logic/Role');

exports.RoleType = require('./Logic/RoleType');

exports.SFJ = require('./Logic/SFJ');

exports.SFJEvaluate = require('./Logic/SFJEvaluate');

exports.SFJLabel = require('./Logic/SFJLabel');

exports.SFJ_Prod = require('./Logic/SFJ_Prod');

exports.SFJ_User = require('./Logic/SFJ_User');

exports.S_User = require('./Logic/S_User');

exports.Shop = require('./Logic/Shop');

exports.ShopActive = require('./Logic/ShopActive');

exports.ShopEvaluate = require('./Logic/ShopEvaluate');

exports.ShopLabel = require('./Logic/ShopLabel');

exports.SiteLabel = require('./Logic/SiteLabel');

exports.SiteProd = require('./Logic/SiteProd');

exports.SiteTrade = require('./Logic/SiteTrade');

exports.SiteUrl = require('./Logic/SiteUrl');

exports.SiteUser = require('./Logic/SiteUser');

exports.SpaceViewRecord = require('./Logic/SpaceViewRecord');

exports.SysMsg = require('./Logic/SysMsg');

exports.Tabel_ID = require('./Logic/Tabel_ID');

exports.TipOff = require('./Logic/TipOff');

exports.TmallTmCat = require('./Logic/TmallTmCat');

exports.UserCollection = require('./Logic/UserCollection');

exports.UserCount = require('./Logic/UserCount');

exports.UserInCome = require('./Logic/UserInCome');

exports.UserLabel = require('./Logic/UserLabel');

exports.UserLevel = require('./Logic/UserLevel');

exports.UserMoney = require('./Logic/UserMoney');

exports.UserPay = require('./Logic/UserPay');

exports.UserPoint = require('./Logic/UserPoint');

exports.UserPreference = require('./Logic/UserPreference');

exports.UserSpace = require('./Logic/UserSpace');

exports.UserTrade = require('./Logic/UserTrade');

exports.VerificationCode = require('./Logic/VerificationCode');

exports.system_config = require('./Logic/system_config');
