// WebTools/lib/Date.js
/**
 * Created with JetBrains WebStorm.
 * User: tianling
 * Date: 13-9-24
 * Time: 下午2:25
 * To change this template use File | Settings | File Templates.
 */

exports.Now = function () {
    var now = new Date ();
    return DateToFullDateTimeString ( now );
}

/**
 * 将ISO时间转换为长时间字符串 格式为：（yyyy-MM-dd hh:mm:ss）
 * @param date
 * @returns {string}
 * @constructor
 */
var DateToFullDateTimeString = function ( date ) {

    var year = date.getFullYear ();
    var month = date.getMonth ();
    var day = date.getDate ();
    var hour = date.getHours ();
    var minute = date.getMinutes ();
    var second = date.getSeconds ();

    var datestr;

    if ( month < 9 ) {
        month = '0' + (month + 1);
    }
    if ( day < 10 ) {
        day = '0' + day;
    }
    if ( hour < 10 ) {
        hour = '0' + hour;
    }
    if ( minute < 10 ) {
        minute = '0' + minute;
    }
    if ( second < 10 ) {
        second = '0' + second;
    }

    datestr = year + '-' + month + '-' + day + ' ' + hour + ':' + minute + ':' + second;
    return datestr;
}


exports.DateToFullDateTimeString = DateToFullDateTimeString;

/**
 * 将ISO时间转换为短时间字符串 格式为：（MM-dd hh:mm）
 * @param date
 * @returns {string}
 * @constructor
 */
var DateToShortDateTimeString = function ( date ) {

    var type1 = typeof date;
    if ( type1 == 'string' )
        date = stringToDate ( date );
    var year = date.getFullYear ();
    var month = date.getMonth ();
    var day = date.getDate ();
    var hour = date.getHours ();
    var minute = date.getMinutes ();
    var second = date.getSeconds ();

    var datestr;

    if ( month < 9 ) {
        month = '0' + (month + 1);
    }
    if ( day < 10 ) {
        day = '0' + day;
    }
    if ( hour < 10 ) {
        hour = '0' + hour;
    }
    if ( minute < 10 ) {
        minute = '0' + minute;
    }
    if ( second < 10 ) {
        second = '0' + second;
    }

    datestr = month + '-' + day + ' ' + hour + ':' + minute;
    return datestr;
}
exports.DateToShortDateTimeString = DateToShortDateTimeString;
/**
 * 时间相减
 * @param  date1  时间1 可以为Date或String
 * @param  date2  时间2 可以为Date或String
 * @returns {number}  结果是秒
 */
exports.dateDiff = function ( date1 , date2 ) {
    var type1 = typeof date1, type2 = typeof date2;
    if ( type1 == 'string' )
        date1 = stringToTime ( date1 );
    else if ( date1.getTime )
        date1 = date1.getTime ();
    if ( type2 == 'string' )
        date2 = stringToTime ( date2 );
    else if ( date2.getTime )
        date2 = date2.getTime ();
    return (date1 - date2) / 1000;//结果是秒
}

/**
 * 将字符串转换为时间
 * @param {String} string
 * @returns {number}
 */
var stringToTime = function ( string ) {
    var f = string.split ( ' ' , 2 );
    var d = (f[0] ? f[0] : '').split ( '-' , 3 );
    var t = (f[1] ? f[1] : '').split ( ':' , 3 );
    return (new Date (
        parseInt ( d[0] , 10 ) || null ,
        (parseInt ( d[1] , 10 ) || 1) - 1 ,
        parseInt ( d[2] , 10 ) || null ,
        parseInt ( t[0] , 10 ) || null ,
        parseInt ( t[1] , 10 ) || null ,
        parseInt ( t[2] , 10 ) || null
    )).getTime ();
}

/**
 * 将字符串转换为时间
 * @param {String} string
 * @returns {number}
 */
exports.stringToTime = stringToTime;
/**
 * 将字符串转换为日期
 * @param string
 * @returns {Date}
 */
var stringToDate = function ( string ) {
    var f = string.split ( ' ' , 2 );
    var d = (f[0] ? f[0] : '').split ( '-' , 3 );
    var t = (f[1] ? f[1] : '').split ( ':' , 3 );
    return new Date (
        parseInt ( d[0] , 10 ) || null ,
        (parseInt ( d[1] , 10 ) || 1) - 1 ,
        parseInt ( d[2] , 10 ) || null ,
        parseInt ( t[0] , 10 ) || null ,
        parseInt ( t[1] , 10 ) || null ,
        parseInt ( t[2] , 10 ) || null
    );

}
exports.stringToDate =   stringToDate;
/**
 * 格式化时间为（几天前，几分钟前，几小时前）
 * @param {Date} time
 * @returns {String}
 * @constructor
 */
exports.FormatData = function ( time ) {
    var now = new Date ();
    var SpanS = parseInt ( dateDiff ( now , time ) );
    var SpanM = parseInt ( SpanS / 60 );//分钟
    if ( SpanM < 60 ) {
        return SpanM + "分钟前";
    }
    var SpanH = parseInt ( SpanM / 60 );
    if ( SpanH < 24 ) {
        return SpanH + "小时前";
    }
    var SpanD = parseInt ( SpanH / 24 );
    if ( SpanD < 14 ) {
        return SpanD + "天前";
    }

    var SpanW = parseInt ( SpanD / 7 );
    if ( SpanW < 4 ) {
        return SpanW + "周前";
    }
    return DateToShortDateTimeString ( time );
}

/**
 * 根据时间获取时间戳
 * @param DateTime
 * @returns {number}
 * @constructor
 */
exports.GetTimestamp = function ( DateTime ) {
    return Math.round ( DateTime.getTime () / 1000 );
}

/**
 * 根据时间戳获取时间
 * @param Timestamp
 * @returns {string}
 * @constructor
 */
exports.GetDateTimeByTimestamp = function ( Timestamp ) {
    var UTCTime = new Date ( Timestamp * 1000 );
    var RealTime = UTCTime.getFullYear () + "-" + (UTCTime.getMonth () + 1) + "-" + UTCTime.getDate () + " " + UTCTime.getHours () + ":" + UTCTime.getMinutes () + ":" + UTCTime.getSeconds ();
    return RealTime;
}

/**Javascript设置要保留的小数位数，四舍五入。
 *   @param {number} Dight  Dight要格式化的 数字
 *   @param  {number} How  How要保留的小数位数
 * @returns  {number}
 * @constructor
 */
exports.ForDight = function ( Dight , How ) {
    Dight = Math.round ( Dight * Math.pow ( 10 , How ) ) / Math.pow ( 10 , How );
    return Dight;
}
